diff --git a/build.gradle.kts b/build.gradle.kts
index ccf911fc..3d904b21 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -1,5 +1,9 @@
 @file:Suppress("UnstableApiUsage")
 
+import com.android.build.api.variant.FilterConfiguration.FilterType.*
+import java.time.LocalDateTime
+import java.time.Duration;
+
 buildscript {
 	repositories {
 		mavenCentral()
@@ -118,6 +122,18 @@ android {
 	testOptions {
 		animationsDisabled = true
 	}
+
+	val d1 = LocalDateTime.of(2021, 12, 31, 23, 59);
+	val d2 = LocalDateTime.now()
+	val minutes = Duration.between(d1, d2).getSeconds() / 60
+	val DateVersionCode = (2112312359 + minutes).toInt()
+	androidComponents {
+		onVariants { variant ->
+			variant.outputs.forEach { output ->
+				output.versionCode.set(DateVersionCode)
+			}
+		}
+	}
 }
 
 pmd {
