diff --git a/build.gradle b/build.gradle
index 47fb7386..890c0dd7 100644
--- a/build.gradle
+++ b/build.gradle
@@ -95,6 +95,13 @@ android {
 			assets.srcDirs = ['assets']
 		}
 	}
+
+        def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+        android.applicationVariants.all { variant ->
+            variant.outputs.all { output ->
+                output.versionCodeOverride = DateVersionCode
+            }
+        }
 }
 
 pmd {
