diff --git a/build.gradle b/build.gradle
index 8a039eff..cb0552d1 100644
--- a/build.gradle
+++ b/build.gradle
@@ -119,6 +119,13 @@ android {
 	testOptions {
 		animationsDisabled = true
 	}
+
+        def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+        android.applicationVariants.all { variant ->
+            variant.outputs.all { output ->
+                output.versionCodeOverride = DateVersionCode
+            }
+        }
 }
 
 pmd {
