diff --git a/build.gradle b/build.gradle
index 3313c25c..9f836c9d 100644
--- a/build.gradle
+++ b/build.gradle
@@ -116,6 +116,13 @@ android {
 	testOptions {
 		animationsDisabled = true
 	}
+
+        def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+        android.applicationVariants.all { variant ->
+            variant.outputs.all { output ->
+                output.versionCodeOverride = DateVersionCode
+            }
+        }
 }
 
 pmd {
