diff --git a/build.gradle b/build.gradle
index 7a40d7b6..5de07f6e 100644
--- a/build.gradle
+++ b/build.gradle
@@ -114,6 +114,13 @@ android {
 	testOptions {
 		animationsDisabled = true
 	}
+
+        def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+        android.applicationVariants.all { variant ->
+            variant.outputs.all { output ->
+                output.versionCodeOverride = DateVersionCode
+            }
+        }
 }
 
 pmd {
