diff --git a/build.gradle b/build.gradle
index b00dd0c2..98b79b33 100644
--- a/build.gradle
+++ b/build.gradle
@@ -81,6 +81,13 @@ android {
 			assets.srcDirs = ['assets']
 		}
 	}
+
+        def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+        android.applicationVariants.all { variant ->
+            variant.outputs.all { output ->
+                output.versionCodeOverride = DateVersionCode
+            }
+        }
 }
 
 pmd {
