diff --git a/build.gradle b/build.gradle
index a53bbd1..5b0dd79 100644
--- a/build.gradle
+++ b/build.gradle
@@ -81,6 +81,13 @@ android {
             assets.srcDirs = ['assets']
         }
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 pmd {
diff --git a/src/main/AndroidManifest.xml b/src/main/AndroidManifest.xml
index 1c50e08..2518cf2 100644
--- a/src/main/AndroidManifest.xml
+++ b/src/main/AndroidManifest.xml
@@ -18,7 +18,6 @@
 
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
           package="org.quantumbadger.redreader"
-          android:versionCode="89"
           android:versionName="1.9.11"
           android:installLocation="auto">
 
