diff --git a/android/app/build.gradle.kts b/android/app/build.gradle.kts
index 127104d69..e59a90551 100644
--- a/android/app/build.gradle.kts
+++ b/android/app/build.gradle.kts
@@ -1,4 +1,6 @@
 import java.util.Properties
+import java.time.LocalDateTime
+import java.time.Duration
 
 plugins {
     id("com.android.application")
@@ -44,6 +46,11 @@ android {
         versionCode = flutter.versionCode
         versionName = flutter.versionName
         multiDexEnabled = true
+        ndk {
+            abiFilters.apply {
+                add("arm64-v8a")
+            }
+        }
     }
 
     signingConfigs {
@@ -62,7 +69,7 @@ android {
             signingConfig = if (hasValidKeystore) signingConfigs.getByName("release") else null
             ndk {
                 debugSymbolLevel = "SYMBOL_TABLE"
-                abiFilters.addAll(listOf("armeabi-v7a", "arm64-v8a", "x86_64"))
+                abiFilters.addAll(listOf("arm64-v8a"))
             }
         }
     }
@@ -82,6 +89,18 @@ android {
             resValue("string", "app_name", "Boorusama")
         }
     }
+
+    val d1 = LocalDateTime.of(2021, 12, 31, 23, 59);
+    val d2 = LocalDateTime.now()
+    val minutes = Duration.between(d1, d2).getSeconds() / 60
+    val DateVersionCode = (2112312359 + minutes).toInt()
+    androidComponents {
+        onVariants { variant ->
+            variant.outputs.forEach { output ->
+                output.versionCode.set(DateVersionCode)
+            }
+        }
+    }
 }
 
 flutter {
diff --git a/pubspec.lock b/pubspec.lock
index 8e61f0081..eaa363296 100644
--- a/pubspec.lock
+++ b/pubspec.lock
@@ -1494,10 +1494,10 @@ packages:
     dependency: "direct main"
     description:
       name: purchases_flutter
-      sha256: "4b845cd93b411f954969b03360d1eaa24a07dfd0379dead514d35211e16b084b"
+      sha256: "8d142ced6086fac9608eaf467e48fdeee289c45a988b819decf1c3b88e3b54b3"
       url: "https://pub.dev"
     source: hosted
-    version: "9.6.2"
+    version: "9.7.0"
   quiver:
     dependency: transitive
     description:
@@ -1705,10 +1705,10 @@ packages:
     dependency: transitive
     description:
       name: shared_preferences_android
-      sha256: a2608114b1ffdcbc9c120eb71a0e207c71da56202852d4aab8a5e30a82269e74
+      sha256: bd14436108211b0d4ee5038689a56d4ae3620fd72fd6036e113bf1345bc74d9e
       url: "https://pub.dev"
     source: hosted
-    version: "2.4.12"
+    version: "2.4.13"
   shared_preferences_foundation:
     dependency: transitive
     description:
