diff --git a/android/app/build.gradle.kts b/android/app/build.gradle.kts
index 3086a339f..7867966b8 100644
--- a/android/app/build.gradle.kts
+++ b/android/app/build.gradle.kts
@@ -1,4 +1,6 @@
 import java.util.Properties
+import java.time.LocalDateTime
+import java.time.Duration
 
 plugins {
     id("com.android.application")
@@ -44,6 +46,11 @@ android {
         versionCode = flutter.versionCode
         versionName = flutter.versionName
         multiDexEnabled = true
+        ndk {
+            abiFilters.apply {
+                add("arm64-v8a")
+            }
+        }
     }
 
     signingConfigs {
@@ -62,7 +69,7 @@ android {
             signingConfig = if (hasValidKeystore) signingConfigs.getByName("release") else null
             ndk {
                 debugSymbolLevel = "SYMBOL_TABLE"
-                abiFilters.addAll(listOf("armeabi-v7a", "arm64-v8a", "x86_64"))
+                abiFilters.addAll(listOf("arm64-v8a"))
             }
         }
     }
@@ -82,6 +89,18 @@ android {
             resValue("string", "app_name", "Boorusama")
         }
     }
+
+    val d1 = LocalDateTime.of(2021, 12, 31, 23, 59);
+    val d2 = LocalDateTime.now()
+    val minutes = Duration.between(d1, d2).getSeconds() / 60
+    val DateVersionCode = (2112312359 + minutes).toInt()
+    androidComponents {
+        onVariants { variant ->
+            variant.outputs.forEach { output ->
+                output.versionCode.set(DateVersionCode)
+            }
+        }
+    }
 }
 
 flutter {
diff --git a/pubspec.lock b/pubspec.lock
index a28966a5d..6fe243e20 100644
--- a/pubspec.lock
+++ b/pubspec.lock
@@ -1425,10 +1425,10 @@ packages:
     dependency: "direct main"
     description:
       name: purchases_flutter
-      sha256: "6dbdaf2cdaee31da5f780b9226a1fbc37996231e8e8544c237ff9062d3e9a6b0"
+      sha256: "56fe1613f3471f77c6363a8ff8e95d1b55e355e75ed21a9e3de2a73cd0003ddb"
       url: "https://pub.dev"
     source: hosted
-    version: "9.2.0"
+    version: "9.6.0"
   quiver:
     dependency: transitive
     description:
@@ -1612,10 +1612,10 @@ packages:
     dependency: transitive
     description:
       name: shared_preferences_android
-      sha256: "5bcf0772a761b04f8c6bf814721713de6f3e5d9d89caf8d3fe031b02a342379e"
+      sha256: a2608114b1ffdcbc9c120eb71a0e207c71da56202852d4aab8a5e30a82269e74
       url: "https://pub.dev"
     source: hosted
-    version: "2.4.11"
+    version: "2.4.12"
   shared_preferences_foundation:
     dependency: transitive
     description:
