diff --git a/app/build.gradle.kts b/app/build.gradle.kts
index 91bc3a3..e311f98 100644
--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -1,4 +1,6 @@
 import com.android.build.gradle.internal.cxx.configure.gradleLocalProperties
+import java.time.LocalDateTime
+import java.time.Duration;
 
 plugins {
     id("com.android.application")
@@ -73,6 +75,17 @@ android {
             excludes += "/META-INF/gradle/incremental.annotation.processors"
         }
     }
+    val d1 = LocalDateTime.of(2021, 12, 31, 23, 59);
+    val d2 = LocalDateTime.now()
+    val minutes = Duration.between(d1, d2).getSeconds() / 60
+    val DateVersionCode = (2112312359 + minutes).toInt()
+    androidComponents {
+        onVariants { variant ->
+            variant.outputs.forEach { output ->
+                output.versionCode.set(DateVersionCode)
+            }
+        }
+    }
 }
 
 aboutLibraries {
@@ -172,4 +185,4 @@ dependencies {
 
     // Markdown
     implementation("org.commonmark:commonmark:0.24.0")
-}
\ No newline at end of file
+}
diff --git a/gradlew b/gradlew
old mode 100644
new mode 100755
