diff --git a/app-android/app/build.gradle b/app-android/app/build.gradle
index e8e2b5fc3..39b66f156 100644
--- a/app-android/app/build.gradle
+++ b/app-android/app/build.gradle
@@ -63,6 +63,13 @@ android {
 		exclude 'META-INF/LICENSE'
 		exclude 'META-INF/ASL2.0'
 	}
+
+        def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+        android.applicationVariants.all { variant ->
+             variant.outputs.all { output ->
+                  output.versionCodeOverride = DateVersionCode
+             }
+        }
 }
 
 task copyResourcesToClasses(type: Copy) {
