diff --git a/app/build.gradle b/app/build.gradle
index 92c00c37..88c5161a 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -70,6 +70,12 @@ android {
     sourceSets {
         androidTest.assets.srcDirs += files("$projectDir/schemas".toString())
     }
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 ext {
@@ -164,4 +170,4 @@ static String getGitWorkingBranch() {
         // git was not found
         return ""
     }
-}
\ No newline at end of file
+}
