diff --git a/app/build.gradle b/app/build.gradle
index a8fb99021..f1e2b66db 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -89,6 +89,13 @@ android {
     buildFeatures {
         viewBinding true
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 ext {
