diff --git a/app/build.gradle.kts b/app/build.gradle.kts
index c4cd5aee9..e03bad8cb 100644
--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -3,6 +3,9 @@
  * SPDX-License-Identifier: GPL-3.0-or-later
  */
 
+import java.time.LocalDateTime
+import java.time.Duration;
+
 plugins {
     alias(libs.plugins.android.application)
     alias(libs.plugins.jetbrains.kotlin.android)
@@ -117,6 +120,17 @@ android {
             )
         }
     }
+    val d1 = LocalDateTime.of(2021, 12, 31, 23, 59);
+    val d2 = LocalDateTime.now()
+    val minutes = Duration.between(d1, d2).getSeconds() / 60
+    val DateVersionCode = (2112312359 + minutes).toInt()
+    androidComponents {
+            onVariants { variant ->
+                    variant.outputs.forEach { output ->
+                            output.versionCode.set(DateVersionCode)
+                   }
+           }
+    }
 }
 
 ksp {
