diff --git a/app/build.gradle b/app/build.gradle
index 0c511a2b3..df3270305 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -95,6 +95,13 @@ android {
     buildFeatures {
         viewBinding true
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 ext {
