diff --git a/app/build.gradle b/app/build.gradle
index f9b2e9bd9..1e89fe89a 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -111,6 +111,18 @@ android {
                          'META-INF/COPYRIGHT']
         }
     }
+
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 ext {
