diff --git a/app/build.gradle b/app/build.gradle
index fdb5ed651..78ebe217d 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -104,6 +104,17 @@ android {
             excludes += ['META-INF/README.md', 'META-INF/CHANGES']
         }
     }
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 ext {
