diff --git a/app/build.gradle b/app/build.gradle
index 11bab00c5..359af4b31 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -103,6 +103,18 @@ android {
         exclude 'META-INF/README.md'
         exclude 'META-INF/CHANGES'
     }
+
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 ext {
