diff --git a/app/build.gradle b/app/build.gradle
index a9e7f39b9..52772cba2 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -85,6 +85,13 @@ android {
     sourceSets {
         androidTest.assets.srcDirs += files("$projectDir/schemas".toString())
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 ext {
