diff --git a/app/build.gradle b/app/build.gradle
index 1b77756e1..ad52d512d 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -96,6 +96,13 @@ android {
     buildFeatures {
         viewBinding true
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 ext {
