diff --git a/app/build.gradle b/app/build.gradle
index 65a36bde9..eef87d959 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -76,6 +76,13 @@ android {
     sourceSets {
         androidTest.assets.srcDirs += files("$projectDir/schemas".toString())
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 ext {
