diff --git a/app/build.gradle b/app/build.gradle
index d52e8b32c..35852a5f0 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -92,6 +92,18 @@ android {
     buildFeatures {
         viewBinding true
     }
+
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 ext {
