diff --git a/build.gradle b/build.gradle
index 94e4907ced..9019f0d7ea 100644
--- a/build.gradle
+++ b/build.gradle
@@ -209,8 +209,6 @@ android {
 
     buildTypes {
         all {
-            signingConfig releaseSigning
-
             applicationVariants.configureEach { variant ->
                 variant.outputs.configureEach {
                     def channel = variant.productFlavors.last().name
@@ -320,6 +318,17 @@ android {
             manifest.srcFile "quickstep/AndroidManifest.xml"
         }
     }
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 androidComponents {
