diff --git a/app/build.gradle.kts b/app/build.gradle.kts
index d6cb0cc..9c2e847 100644
--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -1,5 +1,9 @@
 @file:Suppress("UnstableApiUsage")
 
+import com.android.build.api.variant.FilterConfiguration.FilterType.*
+import java.time.LocalDateTime
+import java.time.Duration;
+
 plugins {
     id("com.android.application")
     id("org.jetbrains.kotlin.android")
@@ -55,12 +59,6 @@ android {
 
     buildTypes {
         release {
-            signingConfig = if (project.hasProperty("RELEASE_STORE_FILE")) {
-                signingConfigs.getByName("release")
-            } else {
-                signingConfigs.getByName("debug")
-            }
-
             postprocessing {
                 isRemoveUnusedCode = true
                 isObfuscate = false
@@ -97,6 +95,17 @@ android {
     composeOptions {
         kotlinCompilerExtensionVersion = "1.5.1"
     }
+    val d1 = LocalDateTime.of(2021, 12, 31, 23, 59);
+    val d2 = LocalDateTime.now()
+    val minutes = Duration.between(d1, d2).getSeconds() / 60
+    val DateVersionCode = (2112312359 + minutes).toInt()
+    androidComponents {
+        onVariants { variant ->
+            variant.outputs.forEach { output ->
+                output.versionCode.set(DateVersionCode)
+            }
+        }
+    }
 }
 
 dependencies {
