diff --git a/networksurvey/build.gradle b/networksurvey/build.gradle
index 9633df1..3281a41 100644
--- a/networksurvey/build.gradle
+++ b/networksurvey/build.gradle
@@ -41,6 +41,9 @@ android {
         setProperty("archivesBaseName", "$applicationName-$versionName")
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
         testInstrumentationRunnerArguments clearPackageData: 'true'
+        ndk {
+             abiFilters "arm64-v8a"
+        }
     }
     buildTypes {
         debug {
@@ -102,6 +105,17 @@ android {
             versionNameSuffix ""
         }
     }
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 configurations.configureEach {
