diff --git a/app/build.gradle b/app/build.gradle
index e11831c..6ed74bf 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -39,6 +39,12 @@ android {
             versionNameSuffix = "-play"
         }
     }
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
