diff --git a/app/build.gradle.kts b/app/build.gradle.kts
index aea0fbe6..e73c8923 100644
--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -1,5 +1,7 @@
 import com.mikepenz.aboutlibraries.plugin.DuplicateMode
 import com.mikepenz.aboutlibraries.plugin.DuplicateRule
+import java.time.LocalDateTime
+import java.time.Duration
 
 plugins {
     id("convention.android-app")
@@ -9,28 +11,10 @@ plugins {
     alias(libs.plugins.aboutLibraries)
 }
 
-bundletool {
-    signingConfig {
-        storeFile.set(rootProject.layout.projectDirectory.file("keystore.jks"))
-        storePassword.set(providers.environmentVariable("SIGNING_STORE_PASSWORD"))
-        keyAlias.set(providers.environmentVariable("SIGNING_KEY_ALIAS"))
-        keyPassword.set(providers.environmentVariable("SIGNING_KEY_PASSWORD"))
-    }
-}
-
 android {
     namespace = "xyz.malkki.neostumbler"
     compileSdk = 36
 
-    signingConfigs {
-        create("release") {
-            storeFile = file("../keystore.jks")
-            storePassword = providers.environmentVariable("SIGNING_STORE_PASSWORD").orNull
-            keyAlias = providers.environmentVariable("SIGNING_KEY_ALIAS").orNull
-            keyPassword = providers.environmentVariable("SIGNING_KEY_PASSWORD").orNull
-        }
-    }
-
     defaultConfig {
         applicationId = "xyz.malkki.neostumbler"
         versionCode = 43
@@ -86,8 +70,6 @@ android {
         getByName("debug") { applicationIdSuffix = ".dev" }
 
         getByName("release") {
-            signingConfig = signingConfigs.getByName("release")
-
             isDebuggable = false
 
             isMinifyEnabled = true
@@ -99,7 +81,7 @@ android {
                 "altbeacon.pro",
             )
 
-            ndk { abiFilters += listOf("armeabi-v7a", "arm64-v8a") }
+            ndk { abiFilters += listOf("arm64-v8a") }
         }
     }
 
@@ -115,7 +97,7 @@ android {
             dimension = "version"
             isDefault = true
 
-            ndk { abiFilters += listOf("armeabi-v7a", "arm64-v8a", "x86", "x86_64") }
+            ndk { abiFilters += listOf("arm64-v8a") }
         }
     }
 
@@ -152,9 +134,9 @@ android {
             isEnable = project.hasProperty("enableAbiSplit")
 
             reset()
-            include("armeabi-v7a", "arm64-v8a")
+            include("arm64-v8a")
 
-            isUniversalApk = true
+            isUniversalApk = false
         }
     }
 
@@ -163,6 +145,18 @@ android {
 
         lintConfig = projectDir.resolve("lint.xml")
     }
+
+    val d1 = LocalDateTime.of(2021, 12, 31, 23, 59);
+    val d2 = LocalDateTime.now()
+    val minutes = Duration.between(d1, d2).getSeconds() / 60
+    val DateVersionCode = (2112312359 + minutes).toInt()
+    androidComponents {
+        onVariants { variant ->
+            variant.outputs.forEach { output ->
+                output.versionCode.set(DateVersionCode)
+             }
+        }
+    }
 }
 
 tasks.register("printVersionName") {
