diff --git a/app/build.gradle b/app/build.gradle
index 65fcfee..3565f3e 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -13,28 +13,10 @@ plugins {
 
 def DB_SCHEMAS_DIR = "$projectDir/schemas"
 
-bundletool {
-    signingConfig {
-        storeFile = file("../keystore.jks")
-        storePassword = System.getenv("SIGNING_STORE_PASSWORD")
-        keyAlias = System.getenv("SIGNING_KEY_ALIAS")
-        keyPassword = System.getenv("SIGNING_KEY_PASSWORD")
-    }
-}
-
 android {
     namespace 'xyz.malkki.neostumbler'
     compileSdk 36
 
-    signingConfigs {
-        release {
-            storeFile file("../keystore.jks")
-            storePassword System.getenv("SIGNING_STORE_PASSWORD")
-            keyAlias System.getenv("SIGNING_KEY_ALIAS")
-            keyPassword System.getenv("SIGNING_KEY_PASSWORD")
-        }
-    }
-
     defaultConfig {
         applicationId "xyz.malkki.neostumbler"
         minSdk 29
@@ -103,8 +85,6 @@ android {
         }
 
         release {
-            signingConfig signingConfigs.release
-
             debuggable false
 
             minifyEnabled true
@@ -113,7 +93,7 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro', 'altbeacon.pro'
 
             ndk {
-                abiFilters('armeabi-v7a', 'arm64-v8a')
+                abiFilters('arm64-v8a')
             }
         }
     }
@@ -132,7 +112,7 @@ android {
             dimension "version"
 
             ndk {
-                abiFilters('armeabi-v7a', 'arm64-v8a', 'x86', 'x86_64')
+                abiFilters('arm64-v8a')
             }
         }
     }
@@ -169,9 +149,9 @@ android {
             enable project.hasProperty("enableAbiSplit")
 
             reset()
-            include 'armeabi-v7a', 'arm64-v8a'
+            include 'arm64-v8a'
 
-            universalApk true
+            universalApk false
         }
     }
 
@@ -184,6 +164,18 @@ android {
     lint {
         lintConfig file("app/lint.xml")
     }
+
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 tasks.register('printVersionName') {
