diff --git a/app/build.gradle.kts b/app/build.gradle.kts
index ae6f234..a999950 100644
--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -1,6 +1,8 @@
 import kotlin.math.roundToInt
 import org.gradle.api.tasks.testing.logging.TestExceptionFormat
 import org.jetbrains.kotlin.gradle.dsl.JvmTarget
+import java.time.LocalDateTime
+import java.time.Duration
 
 plugins {
     id("com.android.application")
@@ -15,15 +17,6 @@ plugins {
 
 private val DB_SCHEMAS_DIR = "$projectDir/schemas"
 
-bundletool {
-    signingConfig {
-        storeFile = file("../keystore.jks")
-        storePassword = System.getenv("SIGNING_STORE_PASSWORD")
-        keyAlias = System.getenv("SIGNING_KEY_ALIAS")
-        keyPassword = System.getenv("SIGNING_KEY_PASSWORD")
-    }
-}
-
 ksp {
     arg("room.schemaLocation", DB_SCHEMAS_DIR)
     arg("room.incremental", "true")
@@ -33,15 +26,6 @@ android {
     namespace = "xyz.malkki.neostumbler"
     compileSdk = 36
 
-    signingConfigs {
-        create("release") {
-            storeFile = file("../keystore.jks")
-            storePassword = System.getenv("SIGNING_STORE_PASSWORD")
-            keyAlias = System.getenv("SIGNING_KEY_ALIAS")
-            keyPassword = System.getenv("SIGNING_KEY_PASSWORD")
-        }
-    }
-
     defaultConfig {
         applicationId = "xyz.malkki.neostumbler"
         minSdk = 29
@@ -104,8 +88,6 @@ android {
         getByName("debug") { applicationIdSuffix = ".dev" }
 
         getByName("release") {
-            signingConfig = signingConfigs.getByName("release")
-
             isDebuggable = false
 
             isMinifyEnabled = true
@@ -117,7 +99,7 @@ android {
                 "altbeacon.pro",
             )
 
-            ndk { abiFilters += listOf("armeabi-v7a", "arm64-v8a") }
+            ndk { abiFilters += listOf("arm64-v8a") }
         }
     }
 
@@ -132,7 +114,7 @@ android {
         create("full") {
             dimension = "version"
 
-            ndk { abiFilters += listOf("armeabi-v7a", "arm64-v8a", "x86", "x86_64") }
+            ndk { abiFilters += listOf("arm64-v8a") }
         }
     }
 
@@ -171,9 +153,9 @@ android {
             isEnable = project.hasProperty("enableAbiSplit")
 
             reset()
-            include("armeabi-v7a", "arm64-v8a")
+            include("arm64-v8a")
 
-            isUniversalApk = true
+            isUniversalApk = false
         }
     }
 
@@ -184,6 +166,18 @@ android {
     }
 
     lint { lintConfig = file("app/lint.xml") }
+
+    val d1 = LocalDateTime.of(2021, 12, 31, 23, 59);
+    val d2 = LocalDateTime.now()
+    val minutes = Duration.between(d1, d2).getSeconds() / 60
+    val DateVersionCode = (2112312359 + minutes).toInt()
+    androidComponents {
+        onVariants { variant ->
+            variant.outputs.forEach { output ->
+                output.versionCode.set(DateVersionCode)
+            }
+        }
+    }
 }
 
 tasks.register("printVersionName") {
