diff --git a/app/build.gradle b/app/build.gradle
index 590352b..92726e1 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -33,6 +33,17 @@ android {
         // We don't really hit the broken use-cases for now
         disable 'InvalidFragmentVersionForActivityResult'
     }
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
