diff --git a/app/build.gradle b/app/build.gradle
index 8d02597..3fb80eb 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -38,6 +38,17 @@ android {
         aidl true
     }
     namespace 'net.typeblog.shelter'
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
