diff --git a/app/build.gradle b/app/build.gradle
index 040babc..a5b2f6d 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -36,6 +36,12 @@ android {
         // We don't really hit the broken use-cases for now
         disable 'InvalidFragmentVersionForActivityResult'
     }
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
