diff --git a/app/build.gradle b/app/build.gradle
index 35ce379..e5ce6e2 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -42,7 +42,7 @@ android {
     buildToolsVersion = '34.0.0'
     defaultConfig {
         applicationId "net.typeblog.shelter"
-        minSdkVersion 24
+        minSdkVersion 28
         targetSdkVersion 34
         versionCode getVersionCode()
         versionName getVersionName()
@@ -66,6 +66,17 @@ android {
         aidl true
     }
     namespace 'net.typeblog.shelter'
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
@@ -78,8 +89,8 @@ dependencies {
     implementation 'com.google.android.material:material:1.10.0'
     implementation 'androidx.localbroadcastmanager:localbroadcastmanager:1.1.0'
     implementation 'mobi.upod:time-duration-picker:1.1.3'
-    debugImplementation project(path: ':setup-wizard-lib', configuration: 'gingerbreadCompatDebugRuntimeElements')
-    releaseImplementation project(path: ':setup-wizard-lib', configuration: 'gingerbreadCompatReleaseRuntimeElements')
+    debugImplementation project(path: ':setup-wizard-lib', configuration: 'platformDeprecatedDebugRuntimeElements')
+    releaseImplementation project(path: ':setup-wizard-lib', configuration: 'platformDeprecatedReleaseRuntimeElements')
     testImplementation 'junit:junit:4.13.2'
     androidTestImplementation 'androidx.test:runner:1.6.0-alpha04'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.6.0-alpha01'
Submodule libs/SetupWizardLibrary contains modified content
diff --git a/libs/SetupWizardLibrary/library/rules.gradle b/libs/SetupWizardLibrary/library/rules.gradle
index b385cf5..b67e9d5 100644
--- a/libs/SetupWizardLibrary/library/rules.gradle
+++ b/libs/SetupWizardLibrary/library/rules.gradle
@@ -24,13 +24,7 @@ android {
         platformDeprecated {
             dimension 'compat'
             // TODO(yukl): Bump this file to v28 once we can properly test that
-            minSdkVersion 27
-        }
-
-        // Provides backwards compatibility for Gingerbread or above, using support libraries.
-        gingerbreadCompat {
-            dimension 'compat'
-            minSdkVersion 14
+            minSdkVersion 28
         }
     }
 
@@ -46,11 +40,6 @@ android {
             java.srcDirs = ['platform/src']
             res.srcDirs = ['platform/res']
         }
-
-        gingerbreadCompat {
-            java.srcDirs = ['gingerbread/src', 'recyclerview/src']
-            res.srcDirs = ['gingerbread/res', 'recyclerview/res']
-        }
     }
 }
 
@@ -72,7 +61,4 @@ dependencies {
     //      }
     //
     platformDeprecatedImplementation deps['support-annotations']
-
-    gingerbreadCompatImplementation deps['support-appcompat-v7']
-    gingerbreadCompatImplementation deps['support-recyclerview-v7']
 }
diff --git a/libs/SetupWizardLibrary/library/standalone-rules.gradle b/libs/SetupWizardLibrary/library/standalone-rules.gradle
index 2b3ecc8..ec88fa5 100644
--- a/libs/SetupWizardLibrary/library/standalone-rules.gradle
+++ b/libs/SetupWizardLibrary/library/standalone-rules.gradle
@@ -26,9 +26,7 @@ ext {
     //      }
     //
     deps = [
-        'support-annotations': 'androidx.annotation:annotation:1.1.0',
-        'support-appcompat-v7': 'androidx.appcompat:appcompat:1.2.0',
-        'support-recyclerview-v7': 'androidx.recyclerview:recyclerview:1.1.0'
+        'support-annotations': 'androidx.annotation:annotation:1.1.0'
     ]
 }
 
diff --git a/libs/SetupWizardLibrary/library/standalone.gradle b/libs/SetupWizardLibrary/library/standalone.gradle
index 84f3c64..9add55e 100644
--- a/libs/SetupWizardLibrary/library/standalone.gradle
+++ b/libs/SetupWizardLibrary/library/standalone.gradle
@@ -16,5 +16,5 @@
 
 apply from: 'standalone-rules.gradle'
 
-android.compileSdkVersion 26
-android.namespace 'com.android.setupwizardlib'
\ No newline at end of file
+android.compileSdkVersion 28
+android.namespace 'com.android.setupwizardlib'
