diff --git a/app/build.gradle b/app/build.gradle
index f3b0a15..513d989 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -31,6 +31,13 @@ android {
         // We don't need Google App Indexing
         disable 'GoogleAppIndexingWarning'
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
