diff --git a/app/build.gradle b/app/build.gradle
index f515f7d..b628ec5 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -30,7 +30,6 @@ android {
             shrinkResources true
             minifyEnabled true
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
-            signingConfig signingConfigs.release
         }
     }
     applicationVariants.all { variant ->
@@ -49,6 +48,12 @@ android {
             universalApk true
         }
     }
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
