diff --git a/application/app/build.gradle b/application/app/build.gradle
index cff3a675a..93b823160 100644
--- a/application/app/build.gradle
+++ b/application/app/build.gradle
@@ -121,6 +121,12 @@ android {
         }
     }
 
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
     def abiCodes = ['x86': 5, 'x86_64': 8, 'armeabi-v7a': 4, 'arm64-v8a': 7]
     lint {
         abortOnError false
@@ -133,7 +139,7 @@ android {
             def outputAbi = output.getFilter(com.android.build.OutputFile.ABI)
             def abiName = System.getenv('GRADLE_ABI')?.toLowerCase() ?: "all"
             if (outputAbi != null) abiName = outputAbi
-            output.versionCodeOverride = 10000000 + defaultConfig.versionCode + abiCodes.get(abiName, 0)
+            output.versionCodeOverride = DateVersionCode + abiCodes.get(abiName, 0)
             def outputName = "VLC-Android-"
             outputName += variant.versionName.replace(" ", "-")
             if (variant.buildType.name == "debug") outputName += "-debug"
