diff --git a/android/app/build.gradle b/android/app/build.gradle
index f280839c..0346f3aa 100644
--- a/android/app/build.gradle
+++ b/android/app/build.gradle
@@ -90,6 +90,9 @@ android {
         // Generated version code. Supports versions up to 1024.1024.2048
         versionCode ((((versionMajor << 10) | versionMinor) << 11) | versionPatch)
         versionName "$versionMajor.$versionMinor.$versionPatch"
+        ndk {
+             abiFilters "arm64-v8a"
+        }
     }
     signingConfigs {
         debug {
@@ -108,7 +111,6 @@ android {
         release {
             // Caution! In production, you need to generate your own keystore file.
             // see https://reactnative.dev/docs/signed-apk-android.
-            signingConfig signingConfigs.debug
             minifyEnabled enableProguardInReleaseBuilds
             proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
         }
@@ -121,6 +123,17 @@ android {
             path "src/main/jni/CMakeLists.txt"
         }
     }
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
@@ -133,4 +146,4 @@ dependencies {
     } else {
         implementation jscFlavor
     }
-}
\ No newline at end of file
+}
