diff --git a/app/build.gradle b/app/build.gradle
index 16f2b88..2cca25f 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -17,7 +17,7 @@ android {
         applicationId "streetwalrus.usbmountr"
         minSdkVersion 19
         targetSdkVersion 28
-        versionCode 8
+        versionCode Integer.parseInt(new Date().format("yyMMddHHmm"))
         versionName getVersionName()
     }
     buildTypes {
