diff --git a/app/build.gradle b/app/build.gradle
index e2f384f8..f8498d6c 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -36,6 +36,12 @@ android {
         sourceCompatibility = '1.8'
         targetCompatibility = '1.8'
     }
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
