diff --git a/mobile/build.gradle b/mobile/build.gradle
index 84c2166..22e51a6 100644
--- a/mobile/build.gradle
+++ b/mobile/build.gradle
@@ -28,22 +28,13 @@ android {
         applicationId 'saschpe.contactevents'
         minSdkVersion 17
         targetSdkVersion 27
-        versionCode 160177
+        versionCode Integer.parseInt(new Date().format("yyMMddHHmm"))
         versionName '1.7.7'
         archivesBaseName = "${applicationId}-${versionName}"
         testInstrumentationRunner 'android.support.test.runner.AndroidJUnitRunner'
         vectorDrawables.useSupportLibrary = true
     }
 
-    signingConfigs {
-        config {
-            keyAlias localProperties["signing.key.alias"]
-            keyPassword localProperties["signing.key.password"]
-            storeFile file(localProperties["signing.store.file"])
-            storePassword localProperties["signing.store.password"]
-        }
-    }
-
     buildTypes {
         debug {
             applicationIdSuffix '.debug'
@@ -51,7 +42,6 @@ android {
         release {
             minifyEnabled true
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
-            signingConfig signingConfigs.config
         }
     }
 }
