diff --git a/mobile/build.gradle b/mobile/build.gradle
index 46af87d..31a4e56 100644
--- a/mobile/build.gradle
+++ b/mobile/build.gradle
@@ -40,15 +40,6 @@ android {
         vectorDrawables.useSupportLibrary = true
     }
 
-    signingConfigs {
-        config {
-            keyAlias localProperties["signing.key.alias"]
-            keyPassword localProperties["signing.key.password"]
-            storeFile file(localProperties["signing.store.file"])
-            storePassword localProperties["signing.store.password"]
-        }
-    }
-
     buildTypes {
         debug {
             applicationIdSuffix '.debug'
@@ -56,7 +47,13 @@ android {
         release {
             minifyEnabled true
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
-            signingConfig signingConfigs.config
+        }
+    }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
         }
     }
 }
