diff --git a/app/build.gradle.kts b/app/build.gradle.kts
index f874b18..08156b0 100644
--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -1,3 +1,6 @@
+import java.time.LocalDateTime
+import java.time.Duration
+
 plugins {
     alias(libs.plugins.android.application)
     alias(libs.plugins.kotlin.android)
@@ -18,9 +21,13 @@ android {
 
     defaultConfig {
         applicationId = "com.dimowner.audiorecorder"
+		val d1 = LocalDateTime.of(2021, 12, 31, 23, 59);
+		val d2 = LocalDateTime.now()
+		val minutes = Duration.between(d1, d2).getSeconds() / 60
+		val DateVersionCode = (2112312359 + minutes).toInt()
         minSdk = 26
         targetSdk = 37
-        versionCode = 942
+        versionCode = DateVersionCode
         versionName = "2.0.2"
 
         testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
@@ -68,7 +75,6 @@ android {
         }
         create("releaseConfig") {
             dimension = "default"
-            signingConfig = signingConfigs.getByName("dev")
             applicationId = "com.dimowner.audiorecorder"
         }
     }
