diff --git a/app/build.gradle b/app/build.gradle
index 769d2d4..98b9c85 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -54,13 +54,19 @@ android {
 			minifyEnabled true
 			shrinkResources true
 			proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
-			signingConfig signingConfigs.release
 		}
 	}
 
 	sourceSets {
 		main.java.srcDirs += 'src/main/kotlin'
 	}
+
+        def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+        android.applicationVariants.all { variant ->
+            variant.outputs.all { output ->
+                output.versionCodeOverride = DateVersionCode
+            }
+        }
 }
 
 kapt {
