diff --git a/app/build.gradle b/app/build.gradle
index 97b610b7..34ec4bd3 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -70,6 +70,13 @@ android {
     sourceSets {
         main.java.srcDirs += 'src/main/kotlin'
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 kapt {
