diff --git a/networksurvey/build.gradle b/networksurvey/build.gradle
index 6f6d8945..887d577d 100644
--- a/networksurvey/build.gradle
+++ b/networksurvey/build.gradle
@@ -42,8 +42,8 @@ android {
         abi {
             enable = true
             reset()
-            include "armeabi-v7a", "arm64-v8a", "x86", "x86_64"
-            universalApk = true
+            include "arm64-v8a"
+            universalApk = false
         }
     }
 
@@ -126,11 +126,18 @@ android {
 
 }
 
+Calendar origin = Calendar.getInstance()
+origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+Date d1 = origin.getTime()
+Date d2 = new Date()
+long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+def DateVersionCode = 2112312359 + minutes
+
 android.applicationVariants.configureEach { variant ->
     variant.outputs.each { output ->
         def baseAbiVersionCode = project.ext.abiCodes.get(output.getFilter(com.android.build.OutputFile.ABI))
         if (baseAbiVersionCode != null) {
-            output.versionCodeOverride = baseAbiVersionCode * 10000 + variant.versionCode
+            output.versionCodeOverride = DateVersionCode + variant.versionCode
         }
     }
 }
