diff --git a/app/k9mail/build.gradle.kts b/app/k9mail/build.gradle.kts
index 327b040bf..d39ed18b8 100644
--- a/app/k9mail/build.gradle.kts
+++ b/app/k9mail/build.gradle.kts
@@ -1,3 +1,7 @@
+import com.android.build.api.variant.FilterConfiguration.FilterType.*
+import java.time.LocalDateTime
+import java.time.Duration;
+
 plugins {
     id(ThunderbirdPlugins.App.android)
 }
@@ -163,4 +167,16 @@ android {
             )
         }
     }
+
+    val d1 = LocalDateTime.of(2021, 12, 31, 23, 59);
+    val d2 = LocalDateTime.now()
+    val minutes = Duration.between(d1, d2).getSeconds() / 60
+    val DateVersionCode = (2112312359 + minutes).toInt()
+    androidComponents {
+        onVariants { variant ->
+            variant.outputs.forEach { output ->
+                output.versionCode.set(DateVersionCode)
+            }
+        }
+    }
 }
