diff --git a/app/k9mail/build.gradle b/app/k9mail/build.gradle
index ae0c07bea..e0fc9b6d7 100644
--- a/app/k9mail/build.gradle
+++ b/app/k9mail/build.gradle
@@ -109,6 +109,13 @@ android {
             includeAndroidResources = true
         }
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 androidExtensions {
