diff --git a/app/k9mail/build.gradle b/app/k9mail/build.gradle
index e45c78f45..a096f3d71 100644
--- a/app/k9mail/build.gradle
+++ b/app/k9mail/build.gradle
@@ -141,6 +141,18 @@ android {
         includeInApk = false
         includeInBundle = false
     }
+
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 if (project.hasProperty('keyAlias')) {
