diff --git a/app/k9mail/build.gradle b/app/k9mail/build.gradle
index 81df0b2c3..bd647383e 100644
--- a/app/k9mail/build.gradle
+++ b/app/k9mail/build.gradle
@@ -110,6 +110,13 @@ android {
             includeAndroidResources = true
         }
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 androidExtensions {
