diff --git a/app/k9mail-jmap/build.gradle b/app/k9mail-jmap/build.gradle
index bcdcc9896..cb9e25d5d 100644
--- a/app/k9mail-jmap/build.gradle
+++ b/app/k9mail-jmap/build.gradle
@@ -122,6 +122,13 @@ android {
             includeAndroidResources = true
         }
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 androidExtensions {
diff --git a/app/k9mail/build.gradle b/app/k9mail/build.gradle
index f2da66043..20c837f99 100644
--- a/app/k9mail/build.gradle
+++ b/app/k9mail/build.gradle
@@ -132,6 +132,13 @@ android {
             includeAndroidResources = true
         }
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 androidExtensions {
