diff --git a/build.gradle b/build.gradle
index 25f5f9e0..2c0f345e 100644
--- a/build.gradle
+++ b/build.gradle
@@ -68,6 +68,12 @@ android {
             minifyEnabled true
         }
     }
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 /**
