diff --git a/build.gradle b/build.gradle
index c1145f5e..439abd74 100644
--- a/build.gradle
+++ b/build.gradle
@@ -87,6 +87,17 @@ android {
         abortOnError false
         checkReleaseBuilds false
     }
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 /**
