diff --git a/build.gradle b/build.gradle
index 8c3074c7..cb4f8985 100644
--- a/build.gradle
+++ b/build.gradle
@@ -84,6 +84,13 @@ android {
             shrinkResources true
         }
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 /**
