diff --git a/build.gradle b/build.gradle
index e4a9b9df..7cf8f6e2 100644
--- a/build.gradle
+++ b/build.gradle
@@ -87,6 +87,13 @@ android {
             shrinkResources true
         }
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 /**
