diff --git a/build.gradle b/build.gradle
index d8debc4a..3c927c95 100644
--- a/build.gradle
+++ b/build.gradle
@@ -74,6 +74,12 @@ android {
             minifyEnabled true
         }
     }
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 /**
