diff --git a/build.gradle b/build.gradle
index 4170fc36..a524202a 100644
--- a/build.gradle
+++ b/build.gradle
@@ -90,6 +90,13 @@ android {
             shrinkResources true
         }
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 /**
