diff --git a/build.gradle b/build.gradle
index 556e0877..c3538654 100644
--- a/build.gradle
+++ b/build.gradle
@@ -92,6 +92,13 @@ android {
             shrinkResources true
         }
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 /**
