diff --git a/build.gradle b/build.gradle
index d7e418ee..77ac4b52 100644
--- a/build.gradle
+++ b/build.gradle
@@ -79,6 +79,12 @@ android {
             minifyEnabled true
         }
     }
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 /**
