diff --git a/app/build.gradle b/app/build.gradle
index 30124a4..dd12f5b 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -91,6 +91,13 @@ android {
             includeAndroidResources = true
         }
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
