diff --git a/app/build.gradle b/app/build.gradle
index e98c590..d09053e 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -39,11 +39,17 @@ android {
                 if (new File(project.projectDir, "src/main/jniLibs/${abi}").exists())
                     include(abi)
             }
-            universalApk true // build an APK with all ABIs too
+            universalApk false // build an APK with all ABIs too
         }
     }
 }
 
+Calendar origin = Calendar.getInstance()
+origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+Date d1 = origin.getTime()
+Date d2 = new Date()
+long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+def DateVersionCode = 2112312359 + minutes
 // Map versionCode so each ABI gets a different one
 // e.g. x86 with version 21 gets a versionCode of 3021
 import com.android.build.VariantOutput
@@ -52,7 +58,7 @@ android.applicationVariants.all { variant ->
         def base = project.ext.abiCodes.get(output.getFilter(VariantOutput.ABI))
         // universal APK keeps the original versionCode
         if (base != null)
-            output.versionCodeOverride = base * 1000 + variant.versionCode
+            output.versionCodeOverride = base * 1000 + variant.versionCode + DateVersionCode
     }
 }
 
diff --git a/buildscripts/buildall.sh b/buildscripts/buildall.sh
index 77f8d34..4d030d4 100755
--- a/buildscripts/buildall.sh
+++ b/buildscripts/buildall.sh
@@ -161,6 +161,5 @@ setup_prefix
 build $target
 
 [ "$target" == "mpv-android" ] && \
-	ls -lh ../app/build/outputs/apk/{default,api29}/*/*.apk
 
 exit 0
diff --git a/buildscripts/scripts/mpv-android.sh b/buildscripts/scripts/mpv-android.sh
index e8ae0c8..d8f4d02 100755
--- a/buildscripts/scripts/mpv-android.sh
+++ b/buildscripts/scripts/mpv-android.sh
@@ -30,7 +30,7 @@ prefix_x86=$(nativeprefix "x86")
 
 PREFIX=$BUILD/prefix/armv7l PREFIX64=$prefix64 PREFIX_X64=$prefix_x64 PREFIX_X86=$prefix_x86 \
 ndk-build -C app/src/main -j$cores
-./gradlew assembleDebug assembleRelease
+#./gradlew assembleRelease
 
 if [ -n "${ANDROID_SIGNING_KEY:-}" ]; then
 	cd "${MPV_ANDROID}/app/build/outputs/apk"
