diff --git a/app/build.gradle b/app/build.gradle
index 17acd15..6872ba6 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -17,6 +17,12 @@ android {
             jni.srcDirs = []
         }
     }
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
diff --git a/buildscripts/buildall.sh b/buildscripts/buildall.sh
index b33a7cb..abfade4 100755
--- a/buildscripts/buildall.sh
+++ b/buildscripts/buildall.sh
@@ -123,6 +123,5 @@ setup_prefix
 build $target
 
 [ "$target" == "mpv-android" ] && \
-	ls -lh ../app/build/outputs/apk/{debug,release}/*.apk
 
 exit 0
diff --git a/buildscripts/scripts/mpv-android.sh b/buildscripts/scripts/mpv-android.sh
index 696a25c..253a38a 100755
--- a/buildscripts/scripts/mpv-android.sh
+++ b/buildscripts/scripts/mpv-android.sh
@@ -30,7 +30,7 @@ prefix_x86=$(nativeprefix "x86")
 
 PREFIX=$BUILD/prefix/armv7l PREFIX64=$prefix64 PREFIX_X64=$prefix_x64 PREFIX_X86=$prefix_x86 \
 ndk-build -C app/src/main -j$cores
-./gradlew assembleDebug assembleRelease
+#./gradlew assembleDebug assembleRelease
 
 if [ -n "${ANDROID_SIGNING_KEY:-}" ]; then
 	cd "${MPV_ANDROID}/app/build/outputs"
