diff --git a/app/build.gradle b/app/build.gradle
index 4b4db69..2cd4c01 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -24,7 +24,7 @@ android {
         applicationId "com.orpheusdroid.screenrecorder"
         minSdkVersion 22
         targetSdkVersion 28
-        versionCode 33
+        versionCode Integer.parseInt(new Date().format("yyMMddHHmm"))
         versionName "2.0.3"
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
     }
