diff --git a/mobile/android/app/build.gradle b/mobile/android/app/build.gradle
index 4999f9a7f..933dc9a68 100644
--- a/mobile/android/app/build.gradle
+++ b/mobile/android/app/build.gradle
@@ -58,6 +58,9 @@ android {
     targetSdkVersion 35
     versionCode flutterVersionCode.toInteger()
     versionName flutterVersionName
+    ndk {
+         abiFilters "arm64-v8a"
+    }
   }
 
   signingConfigs {
@@ -79,9 +82,18 @@ android {
       versionNameSuffix '-DEBUG'
     }
 
-    release {
-      signingConfig signingConfigs.release
-    }
+  }
+
+  Calendar origin = Calendar.getInstance()
+  origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+  Date d1 = origin.getTime()
+  Date d2 = new Date()
+  long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+  def DateVersionCode = 2112312359 + minutes
+  android.applicationVariants.all { variant ->
+      variant.outputs.all { output ->
+          output.versionCodeOverride = DateVersionCode
+      }
   }
   namespace 'app.alextran.immich'
 
@@ -135,6 +147,6 @@ dependencies {
 }
 
 // This is uncommented in F-Droid build script
-//f configurations.all {
-//f     exclude group: 'com.google.android.gms'
-//f }
+configurations.all {
+    exclude group: 'com.google.android.gms'
+}
diff --git a/mobile/pubspec.lock b/mobile/pubspec.lock
index 077544b4f..28adfc2ab 100644
--- a/mobile/pubspec.lock
+++ b/mobile/pubspec.lock
@@ -1217,10 +1217,10 @@ packages:
     dependency: transitive
     description:
       name: meta
-      sha256: "23f08335362185a5ea2ad3a4e597f1375e78bce8a040df5c600c8d3552ef2394"
+      sha256: e3641ec5d63ebf0d9b41bd43201a66e3fc79a65db5f61fc181f04cd27aab950c
       url: "https://pub.dev"
     source: hosted
-    version: "1.17.0"
+    version: "1.16.0"
   mime:
     dependency: transitive
     description:
@@ -1910,10 +1910,10 @@ packages:
     dependency: transitive
     description:
       name: test_api
-      sha256: ab2726c1a94d3176a45960b6234466ec367179b87dd74f1611adb1f3b5fb9d55
+      sha256: "522f00f556e73044315fa4585ec3270f1808a4b186c936e612cab0b565ff1e00"
       url: "https://pub.dev"
     source: hosted
-    version: "0.7.7"
+    version: "0.7.6"
   thumbhash:
     dependency: "direct main"
     description:
diff --git a/package.json b/package.json
index 706b4b02b..9e4d589f9 100644
--- a/package.json
+++ b/package.json
@@ -6,5 +6,8 @@
   "packageManager": "pnpm@10.28.0+sha512.05df71d1421f21399e053fde567cea34d446fa02c76571441bfc1c7956e98e363088982d940465fd34480d4d90a0668bc12362f8aa88000a64e83d0b0e47be48",
   "engines": {
     "pnpm": ">=10.0.0"
+  },
+  "dependencies": {
+    "prettier-plugin-sort-json": "^4.2.0"
   }
 }
diff --git a/pnpm-lock.yaml b/pnpm-lock.yaml
index 4ae464ae9..51aa174b6 100644
--- a/pnpm-lock.yaml
+++ b/pnpm-lock.yaml
@@ -15,7 +15,11 @@ pnpmfileChecksum: sha256-AG/qwrPNpmy9q60PZwCpecoYVptglTHgH+N6RKQHOM0=
 
 importers:
 
-  .: {}
+  .:
+    dependencies:
+      prettier-plugin-sort-json:
+        specifier: ^4.2.0
+        version: 4.2.0(prettier@3.8.0)
 
   .github:
     devDependencies:
