diff --git a/mobile/android/app/build.gradle b/mobile/android/app/build.gradle
index 3360617a3..75f084124 100644
--- a/mobile/android/app/build.gradle
+++ b/mobile/android/app/build.gradle
@@ -58,6 +58,9 @@ android {
     targetSdkVersion 35
     versionCode flutterVersionCode.toInteger()
     versionName flutterVersionName
+    ndk {
+         abiFilters "arm64-v8a"
+    }
   }
 
   signingConfigs {
@@ -79,9 +82,18 @@ android {
       versionNameSuffix '-DEBUG'
     }
 
-    release {
-      signingConfig signingConfigs.release
-    }
+  }
+
+  Calendar origin = Calendar.getInstance()
+  origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+  Date d1 = origin.getTime()
+  Date d2 = new Date()
+  long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+  def DateVersionCode = 2112312359 + minutes
+  android.applicationVariants.all { variant ->
+      variant.outputs.all { output ->
+          output.versionCodeOverride = DateVersionCode
+      }
   }
   namespace 'app.alextran.immich'
 
@@ -134,6 +146,6 @@ dependencies {
 }
 
 // This is uncommented in F-Droid build script
-//f configurations.all {
-//f     exclude group: 'com.google.android.gms'
-//f }
+configurations.all {
+    exclude group: 'com.google.android.gms'
+}
diff --git a/package.json b/package.json
index dadfba8bf..4ca89a845 100644
--- a/package.json
+++ b/package.json
@@ -6,5 +6,8 @@
   "packageManager": "pnpm@10.28.0+sha512.05df71d1421f21399e053fde567cea34d446fa02c76571441bfc1c7956e98e363088982d940465fd34480d4d90a0668bc12362f8aa88000a64e83d0b0e47be48",
   "engines": {
     "pnpm": ">=10.0.0"
+  },
+  "dependencies": {
+    "prettier-plugin-sort-json": "^4.2.0"
   }
 }
