diff --git a/app/build.gradle b/app/build.gradle
index aade9be0..7813b764 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -25,6 +25,13 @@ android {
     lintOptions {
         abortOnError false
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
