diff --git a/app/build.gradle b/app/build.gradle
index 20de60ff..9f7057de 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -23,6 +23,13 @@ android {
     lintOptions {
         abortOnError false
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
