diff --git a/app/build.gradle b/app/build.gradle
index a48439673..045c2b608 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -4,6 +4,7 @@ import java.security.MessageDigest
 
 buildscript {
     repositories {
+        mavenLocal()
         google()
         maven {
             url "https://repo1.maven.org/maven2"
@@ -141,10 +142,6 @@ android {
         buildConfigField "String[]", "LANGUAGES", "new String[]{\"" + autoResConfig().collect { s -> s.replace('-r', '_') }.join('", "') + '"}'
         buildConfigField "int", "CANONICAL_VERSION_CODE", "$canonicalVersionCode"
 
-        ndk {
-            abiFilters 'armeabi-v7a', 'arm64-v8a', 'x86', 'x86_64'
-        }
-
         resConfigs autoResConfig()
 
         splits {
@@ -152,7 +149,7 @@ android {
                 enable true
                 reset()
                 include 'armeabi-v7a', 'arm64-v8a', 'x86', 'x86_64'
-                universalApk true
+                universalApk false
             }
         }
 
@@ -248,6 +245,7 @@ android {
         }
     }
 
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
     android.applicationVariants.all { variant ->
         variant.outputs.each { output ->
             output.outputFileName = output.outputFileName.replace(".apk", "-${variant.versionName}.apk")
@@ -256,7 +254,7 @@ android {
 
             if (postFix >= postFixSize) throw new AssertionError("postFix is too large")
 
-            output.versionCodeOverride = canonicalVersionCode * postFixSize + postFix
+            output.versionCodeOverride = DateVersionCode + postFix
         }
     }
 
