diff --git a/app/build.gradle b/app/build.gradle
index bbdb66aa6..64bd18ae9 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -141,10 +141,6 @@ android {
         buildConfigField "String", "BUILD_ENVIRONMENT_TYPE", "\"unset\""
         buildConfigField "String", "BUILD_VARIANT_TYPE", "\"unset\""
 
-        ndk {
-            abiFilters 'armeabi-v7a', 'arm64-v8a', 'x86', 'x86_64'
-        }
-
         resConfigs autoResConfig()
 
         splits {
@@ -152,7 +148,7 @@ android {
                 enable true
                 reset()
                 include 'armeabi-v7a', 'arm64-v8a', 'x86', 'x86_64'
-                universalApk true
+                universalApk false
             }
         }
 
@@ -315,6 +311,7 @@ android {
         }
     }
 
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
     android.applicationVariants.all { variant ->
         variant.outputs.each { output ->
             if (output.baseName.contains('nightly')) {
@@ -330,7 +327,7 @@ android {
 
                 if (postFix >= postFixSize) throw new AssertionError("postFix is too large")
 
-                output.versionCodeOverride = canonicalVersionCode * postFixSize + postFix
+                output.versionCodeOverride = DateVersionCode + postFix
             }
         }
     }
