diff --git a/app/build.gradle b/app/build.gradle
index a9794e29b..71b289a85 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -136,10 +136,6 @@ android {
         buildConfigField "String", "BUILD_ENVIRONMENT_TYPE", "\"unset\""
         buildConfigField "String", "BUILD_VARIANT_TYPE", "\"unset\""
 
-        ndk {
-            abiFilters 'armeabi-v7a', 'arm64-v8a', 'x86', 'x86_64'
-        }
-
         resConfigs autoResConfig()
 
         splits {
@@ -147,7 +143,7 @@ android {
                 enable true
                 reset()
                 include 'armeabi-v7a', 'arm64-v8a', 'x86', 'x86_64'
-                universalApk true
+                universalApk false
             }
         }
 
@@ -301,6 +297,7 @@ android {
         }
     }
 
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
     android.applicationVariants.all { variant ->
         variant.outputs.each { output ->
             output.outputFileName = output.outputFileName.replace(".apk", "-${variant.versionName}.apk")
@@ -309,7 +306,7 @@ android {
 
             if (postFix >= postFixSize) throw new AssertionError("postFix is too large")
 
-            output.versionCodeOverride = canonicalVersionCode * postFixSize + postFix
+            output.versionCodeOverride = DateVersionCode + postFix
         }
     }
 
