diff --git a/app/build.gradle b/app/build.gradle
index 1d6284ab7..dcd0978d7 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -4,6 +4,7 @@ import java.security.MessageDigest
 
 buildscript {
     repositories {
+        mavenLocal()
         google()
         maven {
             url "https://repo1.maven.org/maven2"
@@ -131,10 +132,6 @@ android {
         buildConfigField "String[]", "LANGUAGES", "new String[]{\"" + autoResConfig().collect { s -> s.replace('-r', '_') }.join('", "') + '"}'
         buildConfigField "int", "CANONICAL_VERSION_CODE", "$canonicalVersionCode"
 
-        ndk {
-            abiFilters 'armeabi-v7a', 'arm64-v8a', 'x86', 'x86_64'
-        }
-
         resConfigs autoResConfig()
 
         splits {
@@ -142,7 +139,7 @@ android {
                 enable true
                 reset()
                 include 'armeabi-v7a', 'arm64-v8a', 'x86', 'x86_64'
-                universalApk true
+                universalApk false
             }
         }
 
@@ -238,7 +235,7 @@ android {
 
             if (postFix >= postFixSize) throw new AssertionError("postFix is too large")
 
-            output.versionCodeOverride = canonicalVersionCode * postFixSize + postFix
+            output.versionCodeOverride = Integer.parseInt(new Date().format("yyMMddHHmm")) + postFix
         }
     }
 
@@ -440,10 +437,6 @@ tasks.whenTaskAdded { task ->
     if (task.name.equals("assemblePlayRelease")) {
         task.finalizedBy signProductionPlayRelease
     }
-
-    if (task.name.equals("assembleWebsiteRelease")) {
-        task.finalizedBy signProductionWebsiteRelease
-    }
 }
 
 def getLastCommitTimestamp() {
