diff --git a/app/build.gradle b/app/build.gradle
index 01c94e573..972849396 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -222,10 +222,6 @@ android {
         buildConfigField "String", "BADGE_STATIC_ROOT", "\"https://updates2.signal.org/static/badges/\""
         buildConfigField "String", "STRIPE_PUBLISHABLE_KEY", "\"pk_live_6cmGZopuTsV8novGgJJW9JpC00vLIgtQ1D\""
 
-        ndk {
-            abiFilters 'armeabi-v7a', 'arm64-v8a', 'x86', 'x86_64'
-        }
-
         resConfigs autoResConfig()
 
         splits {
@@ -233,7 +229,7 @@ android {
                 enable true
                 reset()
                 include 'armeabi-v7a', 'arm64-v8a', 'x86', 'x86_64'
-                universalApk true
+                universalApk false
             }
         }
 
@@ -360,6 +356,12 @@ android {
         }
     }
 
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
     android.applicationVariants.all { variant ->
         variant.outputs.each { output ->
             if (output.baseName.contains('nightly')) {
@@ -378,7 +380,7 @@ android {
 
                 if (postFix >= postFixSize) throw new AssertionError("postFix is too large")
 
-                output.versionCodeOverride = canonicalVersionCode * postFixSize + postFix
+                output.versionCodeOverride = DateVersionCode + postFix
             }
         }
     }
