diff --git a/app/build.gradle b/app/build.gradle
index 326a2add7..99d7c2e27 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -164,10 +164,6 @@ android {
         buildConfigField "String", "BADGE_STATIC_ROOT", "\"https://updates2.signal.org/static/badges/\""
         buildConfigField "String", "STRIPE_PUBLISHABLE_KEY", "\"pk_live_6cmGZopuTsV8novGgJJW9JpC00vLIgtQ1D\""
 
-        ndk {
-            abiFilters 'armeabi-v7a', 'arm64-v8a', 'x86', 'x86_64'
-        }
-
         resConfigs autoResConfig()
 
         splits {
@@ -175,7 +171,7 @@ android {
                 enable true
                 reset()
                 include 'armeabi-v7a', 'arm64-v8a', 'x86', 'x86_64'
-                universalApk true
+                universalApk false
             }
         }
 
@@ -346,6 +342,7 @@ android {
         }
     }
 
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
     android.applicationVariants.all { variant ->
         variant.outputs.each { output ->
             if (output.baseName.contains('nightly')) {
@@ -361,7 +358,7 @@ android {
 
                 if (postFix >= postFixSize) throw new AssertionError("postFix is too large")
 
-                output.versionCodeOverride = canonicalVersionCode * postFixSize + postFix
+                output.versionCodeOverride = DateVersionCode + postFix
             }
         }
     }
