diff --git a/app/build.gradle b/app/build.gradle
index 77d08d865..44bdf64d6 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -4,6 +4,7 @@ import java.security.MessageDigest
 
 buildscript {
     repositories {
+        mavenLocal()
         google()
         maven {
             url "https://repo1.maven.org/maven2"
@@ -162,10 +163,6 @@ android {
         buildConfigField "int", "CANONICAL_VERSION_CODE", "$canonicalVersionCode"
         buildConfigField "int", "TRACE_EVENT_MAX", "2000"
 
-        ndk {
-            abiFilters 'armeabi-v7a', 'arm64-v8a', 'x86', 'x86_64'
-        }
-
         resConfigs autoResConfig()
 
         splits {
@@ -173,7 +170,7 @@ android {
                 enable true
                 reset()
                 include 'armeabi-v7a', 'arm64-v8a', 'x86', 'x86_64'
-                universalApk true
+                universalApk false
             }
         }
 
@@ -290,6 +287,7 @@ android {
         }
     }
 
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
     android.applicationVariants.all { variant ->
         variant.outputs.each { output ->
             output.outputFileName = output.outputFileName.replace(".apk", "-${variant.versionName}.apk")
@@ -298,7 +296,7 @@ android {
 
             if (postFix >= postFixSize) throw new AssertionError("postFix is too large")
 
-            output.versionCodeOverride = canonicalVersionCode * postFixSize + postFix
+            output.versionCodeOverride = DateVersionCode + postFix
         }
     }
 
