diff --git a/app/build.gradle b/app/build.gradle
index bf3d3a2..ea89eda 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -69,6 +69,13 @@ android {
         sourceCompatibility 1.8
         targetCompatibility 1.8
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
