diff --git a/app/build.gradle b/app/build.gradle
index 0486c4b..d73c750 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -64,6 +64,13 @@ android {
         sourceCompatibility 1.8
         targetCompatibility 1.8
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
