diff --git a/app/build.gradle b/app/build.gradle
index cf317e5a..4ca7493b 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -72,6 +72,18 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
+
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 buildscript {
@@ -189,4 +201,4 @@ if (propFile.canRead()) {
 } else {
     println "signing.properties not found"
     android.buildTypes.release.signingConfig = null
-}
\ No newline at end of file
+}
