diff --git a/app/build.gradle b/app/build.gradle
index 8dadef51..0598f1a2 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -71,6 +71,12 @@ android {
     lintOptions {
         warning "MissingTranslation", "GetLocales", "VectorDrawableCompat"
     }
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 buildscript {
