diff --git a/app/build.gradle b/app/build.gradle
index d4af7fff..d070d7e9 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -73,6 +73,13 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 buildscript {
@@ -190,4 +197,4 @@ if (propFile.canRead()) {
 } else {
     println "signing.properties not found"
     android.buildTypes.release.signingConfig = null
-}
\ No newline at end of file
+}
