diff --git a/app/build.gradle b/app/build.gradle
index 541cb3c4..d1bb3d4a 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -97,6 +97,18 @@ android {
     androidResources {
         generateLocaleConfig true
     }
+
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 kapt {
