diff --git a/app/build.gradle b/app/build.gradle
index 8f42b6e..631e6df 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -67,16 +67,6 @@ android {
         abortOnError false
     }
 
-    signingConfigs {
-        // Set the BLOKADA_* constants in your ~/.gradle/gradle.properties to point to your keystore.
-        release {
-            keyAlias 'blokada'
-            keyPassword BLOKADA_KEY_PASSWORD
-            storeFile file(BLOKADA_KEY_PATH)
-            storePassword BLOKADA_STORE_PASSWORD
-        }
-    }
-
     buildTypes {
         debug {
             // By default it is a 'debuggable' build, signed with a debug key.
@@ -91,7 +81,6 @@ android {
             minifyEnabled false
             useProguard false // We don't use Proguard for now.
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
-            signingConfig signingConfigs.release
         }
 
         official {
@@ -262,6 +251,13 @@ android {
         beta.res.srcDirs += 'src/official/res'
         beta.manifest.srcFile 'src/ui-blokada/AndroidManifest.xml'
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
