diff --git a/app/build.gradle b/app/build.gradle
index 18ec34a9..64a4e4fe 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -264,6 +264,13 @@ android {
         beta.res.srcDirs += 'src/official/res'
         beta.manifest.srcFile 'src/ui-blokada/AndroidManifest.xml'
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
