diff --git a/app/build.gradle b/app/build.gradle
index 4ab1dff61..111938748 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -252,7 +252,13 @@ android.applicationVariants.all { variant ->
         // builds.
 
         println("versionName override: $versionName")
-
+        Calendar origin = Calendar.getInstance()
+        origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+        Date d1 = origin.getTime()
+        Date d2 = new Date()
+        long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+        def DateVersionCode = 2112312359 + minutes
+        project.ext.versionCodes = ['armeabi-v7a': 1, 'arm64-v8a': 2, 'x86': 3, 'x86_64': 4]
         variant.outputs.each { output ->
             def isMozillaOnline = project.hasProperty("mozillaOnline") || gradle.hasProperty("localProperties.mozillaOnline")
             def abi = output.getFilter(OutputFile.ABI)
@@ -265,7 +271,7 @@ android.applicationVariants.all { variant ->
             println("versionCode for $abi = $versionCodeOverride, isMozillaOnline = $isMozillaOnline")
 
             output.versionNameOverride = versionName
-            output.versionCodeOverride = versionCodeOverride
+            output.versionCodeOverride = project.ext.versionCodes.get(output.getFilter(com.android.build.OutputFile.ABI), 0) + DateVersionCode
         }
     }
 
