diff --git a/app/build.gradle b/app/build.gradle
index da5d7bfdf..bd1ecd566 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -251,6 +251,8 @@ android.applicationVariants.all { variant ->
         // same version code. Therefore we need to have different version codes for our ARM and x86
         // builds.
 
+        def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+        project.ext.versionCodes = ['armeabi-v7a': 1, 'arm64-v8a': 2, 'x86': 3, 'x86_64': 4]
         variant.outputs.each { output ->
             def abi = output.getFilter(OutputFile.ABI)
             // We use the same version code generator, that we inherited from Fennec, across all channels - even on
@@ -260,7 +262,7 @@ android.applicationVariants.all { variant ->
             println("versionCode for $abi = $versionCodeOverride")
 
             output.versionNameOverride = versionName
-            output.versionCodeOverride = versionCodeOverride
+            output.versionCodeOverride = project.ext.versionCodes.get(output.getFilter(com.android.build.OutputFile.ABI), 0) + DateVersionCode
         }
     }
 
