diff --git a/.experimenter.yaml b/.experimenter.yaml
index 280988d4f..e8d0a792f 100644
--- a/.experimenter.yaml
+++ b/.experimenter.yaml
@@ -33,8 +33,8 @@ messaging:
       type: string
       description: What should be displayed when a control message is selected.
       enum:
-        - show-next-message
         - show-none
+        - show-next-message
     styles:
       type: json
       description: "A map of styles to configure message appearance.\n"
diff --git a/app/build.gradle b/app/build.gradle
index 3bdb10eaa..8b9269586 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -253,6 +253,13 @@ android.applicationVariants.all { variant ->
 
         println("versionName override: $versionName")
 
+        Calendar origin = Calendar.getInstance()
+        origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+        Date d1 = origin.getTime()
+        Date d2 = new Date()
+        long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+        def DateVersionCode = 2112312359 + minutes
+        project.ext.versionCodes = ['armeabi-v7a': 1, 'arm64-v8a': 2, 'x86': 3, 'x86_64': 4]
         variant.outputs.each { output ->
             def isMozillaOnline = project.hasProperty("mozillaOnline") || gradle.hasProperty("localProperties.mozillaOnline")
             def abi = output.getFilter(OutputFile.ABI)
@@ -265,7 +272,7 @@ android.applicationVariants.all { variant ->
             println("versionCode for $abi = $versionCodeOverride, isMozillaOnline = $isMozillaOnline")
 
             output.versionNameOverride = versionName
-            output.versionCodeOverride = versionCodeOverride
+            output.versionCodeOverride = project.ext.versionCodes.get(output.getFilter(com.android.build.OutputFile.ABI), 0) + DateVersionCode
         }
     }
 
