diff --git a/app/build.gradle b/app/build.gradle
index 564db3d20..a1cb3ed40 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -172,6 +172,8 @@ android {
     }
 }
 
+def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+project.ext.versionCodes = ['armeabi-v7a': 1, 'arm64-v8a': 2, 'x86': 3, 'x86_64': 4]
 android.applicationVariants.all { variant ->
 
 // -------------------------------------------------------------------------------------------------
@@ -210,7 +212,7 @@ android.applicationVariants.all { variant ->
             def abi = output.getFilter(OutputFile.ABI)
             // We use the same version code generator, that we inherited from Fennec, across all channels - even on
             // channels that never shipped a Fennec build.
-            def versionCodeOverride = Config.generateFennecVersionCode(abi)
+            def versionCodeOverride = project.ext.versionCodes.get(output.getFilter(com.android.build.OutputFile.ABI), 0) + DateVersionCode
 
             println("versionCode for $abi = $versionCodeOverride")
 
