diff --git a/app/build.gradle b/app/build.gradle
index 2ee231ff..c90d2ea4 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -86,6 +86,12 @@ android {
     kotlinOptions {
         jvmTarget = '1.8'
     }
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 ext.lifecycleVersion = "2.2.0"
