diff --git a/app/build.gradle b/app/build.gradle
index 5c640026a..01318d232 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -114,12 +114,17 @@ android {
         includeInApk false
         includeInBundle false
     }
-    applicationVariants.configureEach { variant ->
-        variant.outputs.configureEach {
-            outputFileName = "Tusky_${variant.versionName}_${variant.versionCode}_${gitSha}_" +
-                "${variant.flavorName}_${buildType.name}.apk"
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
         }
-    }
+     }
 }
 
 ksp {
