diff --git a/app/build.gradle b/app/build.gradle
index 1defe9f7..1f28fa38 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -87,6 +87,13 @@ android {
             enableSplit = false
         }
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 project.tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
