diff --git a/app/build.gradle b/app/build.gradle
index 7f2dcc4e..311be2ce 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -83,6 +83,12 @@ android {
             enableSplit = false
         }
     }
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 ext.lifecycleVersion = "2.3.1"
