diff --git a/app/build.gradle.kts b/app/build.gradle.kts
index de561f2..3bc63b6 100644
--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -4,6 +4,8 @@ import java.util.Properties
 import java.io.StringReader
 import javax.xml.parsers.DocumentBuilderFactory
 import org.xml.sax.InputSource
+import java.time.LocalDateTime
+import java.time.Duration
 
 plugins {
     alias(libs.plugins.android.application)
@@ -57,7 +59,11 @@ android {
         applicationId = "com.aryan.reader"
         minSdk = 26
         targetSdk = 35
-        versionCode = 53
+        val d1 = LocalDateTime.of(2021, 12, 31, 23, 59);
+        val d2 = LocalDateTime.now()
+        val minutes = Duration.between(d1, d2).getSeconds() / 60
+        val DateVersionCode = (2112312359 + minutes).toInt()
+        versionCode = DateVersionCode
         versionName = "1.0.49"
 
         resourceConfigurations += configuredAppLocaleTags()
@@ -71,6 +77,11 @@ android {
         }
         buildConfigField("boolean", "IS_PRO", "false")
         buildConfigField("boolean", "IS_OFFLINE", "false")
+		ndk {
+			abiFilters.apply {
+				add("arm64-v8a")
+			}
+		}
     }
 
     flavorDimensions += "version"
diff --git a/gradlew b/gradlew
old mode 100644
new mode 100755
