diff --git a/app/build.gradle b/app/build.gradle
index 37c31f3e..d89893f6 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -88,6 +88,17 @@ android {
             exclude group: 'androidx.lifecycle', module: 'lifecycle-viewmodel-ktx'
         }
     }
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 allprojects {
     repositories {
@@ -225,4 +236,4 @@ def getCurrentFlavor() {
 println("Flavor: ${getCurrentFlavor()}")
 if ( getCurrentFlavor() == "playstore" ){
     apply plugin: 'com.google.gms.google-services'
-}
\ No newline at end of file
+}
diff --git a/gradlew b/gradlew
old mode 100644
new mode 100755
