diff --git a/app/build.gradle b/app/build.gradle
index 4804f4692..e4faa0fdf 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -48,6 +48,12 @@ android {
         exclude 'META-INF/proguard/androidx-annotations.pro'
     }
     viewBinding.enabled = true
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 allprojects {
     repositories {
