diff --git a/app/build.gradle b/app/build.gradle
index f815b4ac8c..833fcedc2b 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -167,6 +167,18 @@ android {
             unitTests.returnDefaultValues = true
             animationsDisabled true
         }
+
+        Calendar origin = Calendar.getInstance()
+        origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+        Date d1 = origin.getTime()
+        Date d2 = new Date()
+        long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+        def DateVersionCode = 2112312359 + minutes
+        android.applicationVariants.all { variant ->
+            variant.outputs.all { output ->
+                output.versionCodeOverride = DateVersionCode
+            }
+        }
     }
 
     // adapt structure from Eclipse to Gradle/Android Studio expectations;
