diff --git a/app/build.gradle b/app/build.gradle
index 4e6797d8..d2f2272f 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -30,7 +30,7 @@ android {
             enable true
             reset()
             //noinspection ChromeOsAbiSupport
-            include 'armeabi-v7a', 'arm64-v8a'
+            include 'arm64-v8a'
             universalApk false
         }
     }
@@ -85,6 +85,17 @@ android {
         buildConfig true
     }
 
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
     namespace 'com.cappielloantonio.tempo'
 }
 
