diff --git a/build.gradle b/build.gradle
index ead92e284..b76a7262f 100644
--- a/build.gradle
+++ b/build.gradle
@@ -252,4 +252,12 @@ android {
         exclude 'META-INF/BCKEY.DSA'
         exclude 'META-INF/BCKEY.SF'
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
+
 }
