diff --git a/build.gradle b/build.gradle
index 610bc6e00..3a2974f20 100644
--- a/build.gradle
+++ b/build.gradle
@@ -253,4 +253,12 @@ android {
         exclude 'META-INF/BCKEY.DSA'
         exclude 'META-INF/BCKEY.SF'
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
+
 }
