diff --git a/build.gradle b/build.gradle
index a859793c1..f3f6ffdc1 100644
--- a/build.gradle
+++ b/build.gradle
@@ -102,7 +102,7 @@ android {
 
     splits {
         abi {
-            universalApk true
+            universalApk false
             enable true
         }
     }
@@ -271,15 +271,11 @@ android {
         exclude 'META-INF/BCKEY.SF'
     }
 
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    project.ext.versionCodes = ['armeabi-v7a': 1, 'arm64-v8a': 2, 'x86': 3, 'x86_64': 4]
     android.applicationVariants.all { variant ->
-        variant.outputs.each { output ->
-            def baseAbiVersionCode = project.ext.abiCodes.get(output.getFilter(OutputFile.ABI))
-            if (baseAbiVersionCode != null) {
-                output.versionCodeOverride = (100 * variant.versionCode) + baseAbiVersionCode
-            } else {
-                output.versionCodeOverride = (100 * variant.versionCode)
-            }
+        variant.outputs.all { output ->
+            output.versionCodeOverride = project.ext.versionCodes.get(output.getFilter(com.android.build.OutputFile.ABI), 0) + DateVersionCode
         }
-
     }
 }
