diff --git a/app/build.gradle b/app/build.gradle
index 0e3d4695..0f9b9e43 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -30,6 +30,10 @@ android {
         buildFeatures {
             buildConfig true
         }
+
+        ndk {
+             abiFilters "arm64-v8a"
+        }
     }
 
     buildTypes {
@@ -77,6 +81,17 @@ android {
             ]
         }
     }
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 // Disables GoogleServices tasks for F-Droid variant
