diff --git a/fake-store/build.gradle b/fake-store/build.gradle
index 7f3ba8f..72da030 100644
--- a/fake-store/build.gradle
+++ b/fake-store/build.gradle
@@ -40,7 +40,7 @@ android {
 
     defaultConfig {
         versionName getMyVersionName()
-        versionCode getMyVersionCode()
+        versionCode Integer.parseInt(new Date().format("yyMMddHHmm"))
         minSdkVersion androidMinSdk()
         targetSdkVersion androidTargetSdk()
     }
