diff --git a/mobile/android/app/build.gradle b/mobile/android/app/build.gradle
index 506ee9d1a..893c0122b 100644
--- a/mobile/android/app/build.gradle
+++ b/mobile/android/app/build.gradle
@@ -70,11 +70,19 @@ android {
             applicationIdSuffix '.debug'
             versionNameSuffix '-DEBUG'
         }
-
-        release {
-            signingConfig signingConfigs.release
-        }
     }
+
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+         }
+     }
     namespace 'app.alextran.immich'
 }
 
@@ -101,6 +109,6 @@ dependencies {
 }
 
 // This is uncommented in F-Droid build script
-//f configurations.all {
-//f     exclude group: 'com.google.android.gms'
-//f }
+configurations.all {
+    exclude group: 'com.google.android.gms'
+}
diff --git a/mobile/pubspec.lock b/mobile/pubspec.lock
index 8b43cf33d..e0ad9c56e 100644
--- a/mobile/pubspec.lock
+++ b/mobile/pubspec.lock
@@ -669,12 +669,13 @@ packages:
     source: hosted
     version: "11.1.0"
   geolocator_android:
-    dependency: transitive
+    dependency: "direct overridden"
     description:
-      name: geolocator_android
-      sha256: "7aefc530db47d90d0580b552df3242440a10fe60814496a979aa67aa98b1fd47"
-      url: "https://pub.dev"
-    source: hosted
+      path: geolocator_android
+      ref: floss
+      resolved-ref: "6675758d887aee61b3fd0b024fc6ec81befeb366"
+      url: "https://github.com/Zverik/flutter-geolocator.git"
+    source: git
     version: "4.6.1"
   geolocator_apple:
     dependency: transitive
diff --git a/mobile/pubspec.yaml b/mobile/pubspec.yaml
index bf75dad45..35b1926f5 100644
--- a/mobile/pubspec.yaml
+++ b/mobile/pubspec.yaml
@@ -85,11 +85,11 @@ dependency_overrides:
   meta: ^1.11.0
   # TODO: remove once analyzer override is removed
   analyzer_plugin: ^0.11.3
-#f  geolocator_android:
-#f    git:
-#f      url: https://github.com/Zverik/flutter-geolocator.git
-#f      ref: floss
-#f      path: geolocator_android
+  geolocator_android:
+    git:
+      url: https://github.com/Zverik/flutter-geolocator.git
+      ref: floss
+      path: geolocator_android
 
 dev_dependencies:
   flutter_test:
