diff --git a/mobile/android/app/build.gradle b/mobile/android/app/build.gradle
index 52750232c..2ae1e71ac 100644
--- a/mobile/android/app/build.gradle
+++ b/mobile/android/app/build.gradle
@@ -51,27 +51,22 @@ android {
         versionName flutterVersionName
     }
 
-   signingConfigs {
-       release {
-            def keyAliasVal = System.getenv("ALIAS")
-            def keyPasswordVal = System.getenv("ANDROID_KEY_PASSWORD")
-            def storePasswordVal = System.getenv("ANDROID_STORE_PASSWORD")
-
-            keyAlias keyAliasVal ? keyAliasVal : keystoreProperties['keyAlias']
-            keyPassword keyPasswordVal ? keyPasswordVal : keystoreProperties['keyPassword']
-            storeFile file("../key.jks") ? file("../key.jks") : file(keystoreProperties['storeFile'])
-            storePassword storePasswordVal ? storePasswordVal : keystoreProperties['storePassword']
-       }
-   }
-
     buildTypes {
         debug {
             applicationIdSuffix '.debug'
             versionNameSuffix '-DEBUG'
         }
+    }
 
-        release {
-            signingConfig signingConfigs.release
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
         }
     }
 }
@@ -98,6 +93,6 @@ dependencies {
 }
 
 // This is uncommented in F-Droid build script
-//f configurations.all {
-//f     exclude group: 'com.google.android.gms'
-//f }
+configurations.all {
+    exclude group: 'com.google.android.gms'
+}
diff --git a/mobile/pubspec.lock b/mobile/pubspec.lock
index 1f29bf783..b2eaadd63 100644
--- a/mobile/pubspec.lock
+++ b/mobile/pubspec.lock
@@ -669,13 +669,14 @@ packages:
     source: hosted
     version: "11.1.0"
   geolocator_android:
-    dependency: transitive
+    dependency: "direct overridden"
     description:
-      name: geolocator_android
-      sha256: "7aefc530db47d90d0580b552df3242440a10fe60814496a979aa67aa98b1fd47"
-      url: "https://pub.dev"
-    source: hosted
-    version: "4.6.1"
+      path: geolocator_android
+      ref: floss
+      resolved-ref: be830ce0111e3d21f3a4932fa0a7ac888854582c
+      url: "https://github.com/Zverik/flutter-geolocator.git"
+    source: git
+    version: "4.5.5"
   geolocator_apple:
     dependency: transitive
     description:
diff --git a/mobile/pubspec.yaml b/mobile/pubspec.yaml
index 703736e3d..478cb0f0e 100644
--- a/mobile/pubspec.yaml
+++ b/mobile/pubspec.yaml
@@ -78,11 +78,11 @@ dependency_overrides:
   meta: ^1.11.0
   # TODO: remove once analyzer override is removed
   analyzer_plugin: ^0.11.3
-#f  geolocator_android:
-#f    git:
-#f      url: https://github.com/Zverik/flutter-geolocator.git
-#f      ref: floss
-#f      path: geolocator_android
+  geolocator_android:
+    git:
+      url: https://github.com/Zverik/flutter-geolocator.git
+      ref: floss
+      path: geolocator_android
 
 dev_dependencies:
   flutter_test:
