Submodule e2e/test-assets 8885d6d01..9e3b964b0 (rewind):
diff --git a/e2e/test-assets/.github/FUNDING.yml b/e2e/test-assets/.github/FUNDING.yml
index acbb7c7..c7519a4 100644
--- a/e2e/test-assets/.github/FUNDING.yml
+++ b/e2e/test-assets/.github/FUNDING.yml
@@ -1 +1 @@
-custom: ['https://buy.immich.app', 'https://immich.store']
+custom: ['https://buy.immich.app']
diff --git a/e2e/test-assets/metadata/faces/portrait-orientation-6.jpg b/e2e/test-assets/metadata/faces/portrait-orientation-6.jpg
deleted file mode 100644
index a745c96..0000000
Binary files a/e2e/test-assets/metadata/faces/portrait-orientation-6.jpg and /dev/null differ
diff --git a/mobile/android/app/build.gradle b/mobile/android/app/build.gradle
index 0ec511d9f..933b1b2d8 100644
--- a/mobile/android/app/build.gradle
+++ b/mobile/android/app/build.gradle
@@ -50,6 +50,9 @@ android {
         targetSdkVersion 35
         versionCode flutterVersionCode.toInteger()
         versionName flutterVersionName
+        ndk {
+             abiFilters "arm64-v8a"
+        }
     }
 
    signingConfigs {
@@ -72,7 +75,17 @@ android {
         }
 
         release {
-            signingConfig signingConfigs.release
+        }
+    }
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
         }
     }
     namespace 'app.alextran.immich'
@@ -101,6 +114,6 @@ dependencies {
 }
 
 // This is uncommented in F-Droid build script
-//f configurations.all {
-//f     exclude group: 'com.google.android.gms'
-//f }
+configurations.all {
+    exclude group: 'com.google.android.gms'
+}
