diff --git a/app/build.gradle b/app/build.gradle
index 96397fa2..fd7114ad 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -131,6 +131,18 @@ android {
     textOutput file('stdout')
     textReport true
   }
+
+  Calendar origin = Calendar.getInstance()
+  origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+  Date d1 = origin.getTime()
+  Date d2 = new Date()
+  long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+  def DateVersionCode = 2112312359 + minutes
+  android.applicationVariants.all { variant ->
+      variant.outputs.all { output ->
+          output.versionCodeOverride = DateVersionCode
+      }
+  }
 }
 
 // https://chris.banes.dev/composable-metrics/
