diff --git a/app/build.gradle b/app/build.gradle
index e4aa46b..4e90080 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -110,6 +110,13 @@ android {
   kotlinOptions {
     jvmTarget = "1.8"
   }
+
+  def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+  android.applicationVariants.all { variant ->
+      variant.outputs.all { output ->
+          output.versionCodeOverride = DateVersionCode
+      }
+  }
 }
 
 dependencies {
diff --git a/gradle.properties b/gradle.properties
index 70a1496..551839e 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -19,4 +19,5 @@
 android.enableJetifier=true
 android.useAndroidX=true
 
+org.gradle.jvmargs=-Xmx1g
 org.gradle.caching=true
diff --git a/rome b/rome
index d1eee4a..76d28e4 160000
--- a/rome
+++ b/rome
@@ -1 +1 @@
-Subproject commit d1eee4a63fc321e749e9708cde5c98688320cf29
+Subproject commit 76d28e4d4432f932242cc8df2598bc65efb6056d
