diff --git a/app/build.gradle b/app/build.gradle
index 5b72c24f..2eba6136 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -133,6 +133,18 @@ android {
     exclude 'META-INF/AL2.0'
     exclude 'META-INF/LGPL2.1'
   }
+
+  Calendar origin = Calendar.getInstance()
+  origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+  Date d1 = origin.getTime()
+  Date d2 = new Date()
+  long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+  def DateVersionCode = 2112312359 + minutes
+  android.applicationVariants.all { variant ->
+      variant.outputs.all { output ->
+          output.versionCodeOverride = DateVersionCode
+      }
+  }
 }
 
 configurations.all {
diff --git a/gradle.properties b/gradle.properties
index 8079a17b..d753d76b 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -20,4 +20,5 @@ org.gradle.jvmargs=-Xmx2048m
 android.enableJetifier=true
 android.useAndroidX=true
 
+org.gradle.jvmargs=-Xmx1g
 org.gradle.caching=true
