diff --git a/app/build.gradle b/app/build.gradle
index db657e1c..8e230fd6 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -118,6 +118,13 @@ android {
     exclude 'META-INF/AL2.0'
     exclude 'META-INF/LGPL2.1'
   }
+
+  def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+  android.applicationVariants.all { variant ->
+      variant.outputs.all { output ->
+          output.versionCodeOverride = DateVersionCode
+      }
+  }
 }
 
 configurations.all {
diff --git a/gradle.properties b/gradle.properties
index 70a14964..551839ee 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -19,4 +19,5 @@
 android.enableJetifier=true
 android.useAndroidX=true
 
+org.gradle.jvmargs=-Xmx1g
 org.gradle.caching=true
