diff --git a/app/build.gradle b/app/build.gradle
index 166765ac..1d5a5633 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -110,6 +110,13 @@ android {
   kotlinOptions {
     jvmTarget = "1.8"
   }
+
+  def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+  android.applicationVariants.all { variant ->
+      variant.outputs.all { output ->
+          output.versionCodeOverride = DateVersionCode
+      }
+  }
 }
 
 dependencies {
diff --git a/gradle.properties b/gradle.properties
index 70a14964..551839ee 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -19,4 +19,5 @@
 android.enableJetifier=true
 android.useAndroidX=true
 
+org.gradle.jvmargs=-Xmx1g
 org.gradle.caching=true
