diff --git a/app/build.gradle b/app/build.gradle
index 5b28fec..2ffc49e 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -110,6 +110,13 @@ android {
   kotlinOptions {
     jvmTarget = "1.8"
   }
+
+  def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+  android.applicationVariants.all { variant ->
+      variant.outputs.all { output ->
+          output.versionCodeOverride = DateVersionCode
+      }
+  }
 }
 
 play {
diff --git a/gradle.properties b/gradle.properties
index 70a1496..15fc148 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -18,5 +18,5 @@
 # org.gradle.parallel=true
 android.enableJetifier=true
 android.useAndroidX=true
-
+org.gradle.jvmargs=-Xmx1g
 org.gradle.caching=true
