diff --git a/app/build.gradle b/app/build.gradle
index 5251e764..afa7f0d6 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -97,6 +97,13 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 // keystore -------------------------------------------------
diff --git a/app/build.properties b/app/build.properties
index 0275990a..61219a15 100644
--- a/app/build.properties
+++ b/app/build.properties
@@ -1,7 +1,7 @@
 #Build Properties
-#Sat Nov 30 13:21:04 EST 2019
+#Sun Dec 01 11:49:18 WET 2019
 version_minor=1
 version_build=0
-version_patch=1
+version_patch=2
+version_store=52
 version_major=2
-version_store=51
