diff --git a/app/build.gradle b/app/build.gradle
index a054826e..065d5f07 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -113,6 +113,18 @@ android {
     kotlinOptions {
         jvmTarget = "1.8"
     }
+
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 allOpen {
diff --git a/app/build.properties b/app/build.properties
index f2f6f546..b3a864f6 100644
--- a/app/build.properties
+++ b/app/build.properties
@@ -1,7 +1,7 @@
 #Build Properties
-#Mon Jan 03 10:10:05 EST 2022
+#Tue Jan 04 15:15:23 WET 2022
 version_minor=0
-version_store=58
-version_patch=6
+version_store=61
+version_patch=9
 version_build=0
 version_major=3
