diff --git a/app/build.gradle b/app/build.gradle
index 1feea08c..85aefc67 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -113,6 +113,13 @@ android {
     kotlinOptions {
         jvmTarget = "1.8"
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 allOpen {
diff --git a/app/build.properties b/app/build.properties
index 9d7c14c0..23eb509b 100644
--- a/app/build.properties
+++ b/app/build.properties
@@ -1,7 +1,7 @@
 #Build Properties
-#Sun Mar 28 12:10:57 EDT 2021
+#Sun Mar 28 20:24:38 WEST 2021
 version_minor=0
-version_store=56
-version_patch=4
 version_build=0
+version_patch=5
 version_major=3
+version_store=57
