diff --git a/app/build.gradle b/app/build.gradle
index 77ab33c4..d930d561 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -113,6 +113,13 @@ android {
     kotlinOptions {
         jvmTarget = "1.8"
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 allOpen {
diff --git a/app/build.properties b/app/build.properties
index 001220d4..828a3e13 100644
--- a/app/build.properties
+++ b/app/build.properties
@@ -1,7 +1,7 @@
 #Build Properties
-#Wed Dec 23 10:55:19 EST 2020
+#Wed Dec 23 21:56:34 WET 2020
 version_minor=0
-version_store=55
-version_patch=3
 version_build=0
+version_patch=4
 version_major=3
+version_store=56
