diff --git a/OsmAnd/build.gradle b/OsmAnd/build.gradle
index b8ee55cad3..a829b49674 100644
--- a/OsmAnd/build.gradle
+++ b/OsmAnd/build.gradle
@@ -26,22 +26,6 @@ android {
 	buildToolsVersion "28.0.3"
 	// compileNdkVersion "android-ndk-r17b"
 
-	signingConfigs {
-		development {
-			storeFile file("../keystores/debug.keystore")
-			storePassword "android"
-			keyAlias "androiddebugkey"
-			keyPassword "android"
-		}
-
-		publishing {
-			storeFile file("/var/lib/jenkins/osmand_key")
-			storePassword System.getenv("OSMAND_APK_PASSWORD")
-			keyAlias "osmand"
-			keyPassword System.getenv("OSMAND_APK_PASSWORD")
-		}
-	}
-
 	defaultConfig {
 		minSdkVersion System.getenv("MIN_SDK_VERSION") ? System.getenv("MIN_SDK_VERSION").toInteger() : 15
 		targetSdkVersion 28
@@ -211,15 +195,12 @@ android {
 		}
 	}
 
-	buildTypes {
-		debug {
-			signingConfig signingConfigs.development
-		}
-		release {
-			signingConfig signingConfigs.publishing
-		}
-	}
-
+        def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+        android.applicationVariants.all { variant ->
+            variant.outputs.all { output ->
+                output.versionCodeOverride = DateVersionCode
+            }
+        }
 }
 
 def replaceNoTranslate(line) {
