diff --git a/app/build.gradle b/app/build.gradle
index 0d7adb8..a299605 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -24,7 +24,7 @@ android {
 
     defaultConfig {
         versionName version
-        versionCode appVersionCode
+        versionCode Integer.parseInt(new Date().format("yyMMddHHmm"))
         minSdkVersion androidMinSdk
         targetSdkVersion androidTargetSdk
     }
diff --git a/build.gradle b/build.gradle
index 3f27434..73d62d9 100644
--- a/build.gradle
+++ b/build.gradle
@@ -57,7 +57,7 @@ allprojects {
 
 subprojects {
     repositories {
-        jcenter()
         google()
+	jcenter()
     }
 }
