diff --git a/app/build.gradle b/app/build.gradle
index 2c348c7..0a9239f 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -23,7 +23,7 @@ android {
 
     defaultConfig {
         versionName version
-        versionCode appVersionCode
+        versionCode Integer.parseInt(new Date().format("yyMMddHHmm"))
         minSdkVersion androidMinSdk
         targetSdkVersion androidTargetSdk
     }
diff --git a/build.gradle b/build.gradle
index c74c3ee..bd2fe6a 100644
--- a/build.gradle
+++ b/build.gradle
@@ -52,7 +52,7 @@ allprojects {
 
 subprojects {
     repositories {
-        jcenter()
         google()
+	jcenter()
     }
 }
