diff --git a/fenix/app/build.gradle b/fenix/app/build.gradle
index c473a89958..74c2a5d208 100644
--- a/fenix/app/build.gradle
+++ b/fenix/app/build.gradle
@@ -275,6 +275,13 @@ android.applicationVariants.configureEach { variant ->
         def versionName = variant.buildType.name == 'nightly' ? Config.nightlyVersionName() : Config.releaseVersionName(project)
         println("versionName override: $versionName")
 
+        Calendar origin = Calendar.getInstance()
+        origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+        Date d1 = origin.getTime()
+        Date d2 = new Date()
+        long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+        def DateVersionCode = 2112312359 + minutes
+        project.ext.versionCodes = ['armeabi-v7a': 1, 'arm64-v8a': 2, 'x86': 3, 'x86_64': 4]
         variant.outputs.each { output ->
             def isMozillaOnline = project.hasProperty("mozillaOnline") || gradle.hasProperty("localProperties.mozillaOnline")
             def abi = output.getFilter(FilterConfiguration.FilterType.ABI.name())
@@ -289,7 +296,7 @@ android.applicationVariants.configureEach { variant ->
             if (versionName != null) {
                 output.versionNameOverride = versionName
             }
-            output.versionCodeOverride = versionCodeOverride
+            output.versionCodeOverride = project.ext.versionCodes.get(output.getFilter(com.android.build.OutputFile.ABI), 0) + DateVersionCode
         }
     } else if (gradle.hasProperty("localProperties.branchBuild.fenix.version")) {
         def versionName = gradle.getProperty("localProperties.branchBuild.fenix.version")
