diff --git a/androidApp/build.gradle.kts b/androidApp/build.gradle.kts
index 1afe334..1bca859 100644
--- a/androidApp/build.gradle.kts
+++ b/androidApp/build.gradle.kts
@@ -1,4 +1,7 @@
 import ru.stersh.youamp.AppBuildInfo
+import com.android.build.api.variant.FilterConfiguration.FilterType.*
+import java.time.LocalDateTime
+import java.time.Duration;
 
 plugins {
     alias(libs.plugins.kotlin.multiplatform)
@@ -93,6 +96,17 @@ android {
             isShrinkResources = false
         }
     }
+    val d1 = LocalDateTime.of(2021, 12, 31, 23, 59);
+    val d2 = LocalDateTime.now()
+    val minutes = Duration.between(d1, d2).getSeconds() / 60
+    val DateVersionCode = (2112312359 + minutes).toInt()
+    androidComponents {
+        onVariants { variant ->
+            variant.outputs.forEach { output ->
+                output.versionCode.set(DateVersionCode)
+            }
+        }
+    }
 }
 
 dependencies {
