diff --git a/app/build.gradle b/app/build.gradle
index a49c12c3..e9b895ef 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -65,6 +65,13 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 repositories {
@@ -95,4 +102,4 @@ dependencies {
 
     // Only compile AdMob ads for the free flavor.
     freeImplementation 'com.google.android.gms:play-services-ads:20.0.0'
-}
\ No newline at end of file
+}
