diff --git a/app/build.gradle b/app/build.gradle
index 01b98033..2ab047c9 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -60,6 +60,13 @@ android {
             dimension "basic"
         }
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 repositories {
@@ -90,4 +97,4 @@ dependencies {
 
     // Only compile AdMob ads for the free flavor.
     freeImplementation 'com.google.android.gms:play-services-ads:20.1.0'
-}
\ No newline at end of file
+}
