diff --git a/app/build.gradle b/app/build.gradle
index 5ca89f82..e96e9df6 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -64,6 +64,13 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 repositories {
@@ -94,4 +101,4 @@ dependencies {
 
     // Only compile Firebase ads for the free flavor.
     freeImplementation 'com.google.firebase:firebase-ads:18.3.0'
-}
\ No newline at end of file
+}
