diff --git a/app/build.gradle b/app/build.gradle
index 785bb416..d4663a9a 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -67,6 +67,12 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 repositories {
@@ -97,4 +103,4 @@ dependencies {
 
     // Only compile Firebase ads for the free flavor.
     freeImplementation 'com.google.firebase:firebase-ads:19.0.1'
-}
\ No newline at end of file
+}
