diff --git a/app/build.gradle b/app/build.gradle
index 5ce0872c..9e6e8656 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -62,6 +62,13 @@ android {
             dimension "basic"
         }
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
@@ -86,4 +93,4 @@ dependencies {
 
     // Only compile AdMob ads for the free flavor.
     freeImplementation 'com.google.android.gms:play-services-ads:20.4.0'
-}
\ No newline at end of file
+}
