diff --git a/app/build.gradle b/app/build.gradle
index ff5a8a82..90a4d27b 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -62,6 +62,13 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
@@ -84,4 +91,4 @@ dependencies {
 
     // Only compile Firebase ads for the free flavor.
     freeImplementation 'com.google.firebase:firebase-ads:18.3.0'
-}
\ No newline at end of file
+}
