diff --git a/app/build.gradle b/app/build.gradle
index 4223e3a8..3e194999 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -66,6 +66,13 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 repositories {
@@ -96,4 +103,4 @@ dependencies {
 
     // Only compile AdMob ads for the free flavor.
     freeImplementation 'com.google.android.gms:play-services-ads:19.7.0'
-}
\ No newline at end of file
+}
