diff --git a/app/build.gradle b/app/build.gradle
index ea64888f..1ac2b016 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -127,6 +127,17 @@ android {
     lint {
         lintConfig = file("lint.xml")
     }
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 allOpen {
diff --git a/app/build.properties b/app/build.properties
index 606d01ff..dd7d5f31 100644
--- a/app/build.properties
+++ b/app/build.properties
@@ -1,7 +1,7 @@
 #Build Properties
-#Sat Jan 10 11:59:55 EST 2026
-version_build=43
-version_major=3
+#Sat Jan 10 23:05:36 WET 2026
 version_minor=2
-version_patch=1
-version_store=70
+version_store=71
+version_patch=2
+version_build=0
+version_major=3
