diff --git a/app/build.gradle b/app/build.gradle
index e41e53a1..78fc920b 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -123,6 +123,17 @@ android {
     lint {
         lintConfig = file("lint.xml")
     }
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 allOpen {
diff --git a/app/build.properties b/app/build.properties
index 27c2b0aa..679d7f56 100644
--- a/app/build.properties
+++ b/app/build.properties
@@ -1,7 +1,7 @@
 #Build Properties
-#Wed Jun 11 16:34:06 EDT 2025
-version_build=19
-version_major=3
+#Tue Dec 02 12:45:08 WET 2025
 version_minor=2
-version_patch=0
-version_store=69
+version_store=71
+version_patch=2
+version_build=0
+version_major=3
