diff --git a/ui/build.gradle.kts b/ui/build.gradle.kts
index 52ad0b78..3c015fb5 100644
--- a/ui/build.gradle.kts
+++ b/ui/build.gradle.kts
@@ -1,5 +1,8 @@
 @file:Suppress("UnstableApiUsage")
 
+import java.time.LocalDateTime
+import java.time.Duration;
+
 val pkg: String = providers.gradleProperty("wireguardPackageName").get()
 
 plugins {
@@ -16,11 +19,20 @@ android {
     }
     namespace = pkg
     defaultConfig {
+        val d1 = LocalDateTime.of(2021, 12, 31, 23, 59);
+        val d2 = LocalDateTime.now()
+        val minutes = Duration.between(d1, d2).getSeconds() / 60
+        val DateVersionCode = (2112312359 + minutes).toInt()
         applicationId = pkg
         minSdk = 24
-        versionCode = providers.gradleProperty("wireguardVersionCode").get().toInt()
+        versionCode = DateVersionCode
         versionName = providers.gradleProperty("wireguardVersionName").get()
         buildConfigField("int", "MIN_SDK_VERSION", minSdk.toString())
+        ndk {
+            abiFilters.apply {
+                add("arm64-v8a")
+            }
+        }
     }
     compileOptions {
         sourceCompatibility = JavaVersion.VERSION_17
