diff --git a/play-services-core/build.gradle b/play-services-core/build.gradle
index 03a0b70..0b3862d 100644
--- a/play-services-core/build.gradle
+++ b/play-services-core/build.gradle
@@ -75,7 +75,7 @@ android {
 
     defaultConfig {
         versionName version
-        versionCode appVersionCode
+        versionCode Integer.parseInt(new Date().format("yyMMddHHmm"))
 
         minSdkVersion androidMinSdk
         targetSdkVersion androidTargetSdk
