diff --git a/play-services-core/build.gradle b/play-services-core/build.gradle
index b55d79e..6d4fbdb 100644
--- a/play-services-core/build.gradle
+++ b/play-services-core/build.gradle
@@ -66,7 +66,7 @@ android {
 
     defaultConfig {
         versionName version
-        versionCode appVersionCode
+        versionCode Integer.parseInt(new Date().format("yyMMddHHmm"))
 
         minSdkVersion androidMinSdk
         targetSdkVersion androidTargetSdk
