diff --git a/play-services-core/build.gradle b/play-services-core/build.gradle
index f32042e..d7b3532 100644
--- a/play-services-core/build.gradle
+++ b/play-services-core/build.gradle
@@ -80,7 +80,7 @@ android {
 
     defaultConfig {
         versionName version
-        versionCode appVersionCode
+        versionCode Integer.parseInt(new Date().format("yyMMddHHmm"))
 
         minSdkVersion androidMinSdk
         targetSdkVersion androidTargetSdk
