diff --git a/app/build.gradle b/app/build.gradle
index c8538a34c..767ab16c5 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -75,6 +75,18 @@ android {
     packagingOptions {
         exclude 'META-INF/library_release.kotlin_module'
     }
+
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
