diff --git a/app/build.gradle b/app/build.gradle
index daae7ddcb..f9b7b6d28 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -74,6 +74,13 @@ android {
     packagingOptions {
         exclude 'META-INF/library_release.kotlin_module'
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
