diff --git a/app/build.gradle b/app/build.gradle
index bbaeef0d3..8ff1d439e 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -44,6 +44,10 @@ android {
         buildConfigField("String", "CUSTOM_LOGO_URL", "\"$CUSTOM_LOGO_URL\"")
         buildConfigField("String", "CUSTOM_INSTANCE", "\"$CUSTOM_INSTANCE\"")
         buildConfigField("String", "SUPPORT_ACCOUNT_URL", "\"$SUPPORT_ACCOUNT_URL\"")
+
+        ndk {
+             abiFilters "arm64-v8a"
+        }
     }
     signingConfigs {
         test {
@@ -81,7 +85,6 @@ android {
     flavorDimensions += "color"
     productFlavors {
         blue {
-            signingConfig = signingConfigs.release
         }
         green {
             resValue "string", "app_name", APP_NAME + " Test"
@@ -146,10 +149,16 @@ android {
         includeInApk false
         includeInBundle false
     }
-    applicationVariants.configureEach { variant ->
-        variant.outputs.configureEach {
-            outputFileName = "Tusky_${variant.versionName}_${variant.versionCode}_${gitSha}_" +
-                "${variant.flavorName}_${buildType.name}.apk"
+
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
         }
     }
 
