diff --git a/twidere/build.gradle b/twidere/build.gradle
index 7fce8ac33..50457b682 100644
--- a/twidere/build.gradle
+++ b/twidere/build.gradle
@@ -142,6 +142,13 @@ android {
         exclude 'sdk-version.txt'
         exclude 'build-data.properties'
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 task buildTranslationArray {
@@ -324,4 +331,4 @@ if (enableGoogleVariant) {
 // START Non-FOSS component
     apply plugin: 'com.google.gms.google-services'
 // END Non-FOSS component
-}
\ No newline at end of file
+}
