diff --git a/twidere/build.gradle b/twidere/build.gradle
index 0e41ee86e..2a39270ac 100644
--- a/twidere/build.gradle
+++ b/twidere/build.gradle
@@ -146,6 +146,12 @@ android {
         exclude 'build-data.properties'
     }
 
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
@@ -295,4 +301,4 @@ if (enableGoogleVariant) {
 // START Non-FOSS component
     apply plugin: 'com.google.gms.google-services'
 // END Non-FOSS component
-}
\ No newline at end of file
+}
