diff --git a/twidere/build.gradle b/twidere/build.gradle
index e63e8dd14..77b2e1b6d 100644
--- a/twidere/build.gradle
+++ b/twidere/build.gradle
@@ -142,6 +142,13 @@ android {
         exclude 'sdk-version.txt'
         exclude 'build-data.properties'
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 task buildTranslationArray {
