diff --git a/twidere/build.gradle b/twidere/build.gradle
index 34e307a72..24497cd4b 100644
--- a/twidere/build.gradle
+++ b/twidere/build.gradle
@@ -143,6 +143,12 @@ android {
         exclude 'build-data.properties'
     }
 
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 repositories {
@@ -279,4 +285,4 @@ if (enableGoogleVariant) {
 // START Non-FOSS component
     apply plugin: 'com.google.gms.google-services'
 // END Non-FOSS component
-}
\ No newline at end of file
+}
