diff --git a/app/build.gradle b/app/build.gradle
index 8100c50..1efbc66 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -43,6 +43,18 @@ android {
     packagingOptions.resources {
         excludes += ["DebugProbesKt.bin", "META-INF/**.version", "kotlin/**.kotlin_builtins", "kotlin-tooling-metadata.json"]
     }
+
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile) {
@@ -69,4 +81,4 @@ dependencies {
 
     implementation "com.github.bumptech.glide:glide:4.15.1"
     implementation "com.davemorrissey.labs:subsampling-scale-image-view-androidx:3.10.0"
-}
\ No newline at end of file
+}
diff --git a/app/src/main/res/values-nl/strings.xml b/app/src/main/res/values-nl/strings.xml
index f5fddcd..02bc030 100644
--- a/app/src/main/res/values-nl/strings.xml
+++ b/app/src/main/res/values-nl/strings.xml
@@ -140,8 +140,6 @@
     <string name="cant_load_image">Afbeelding kan niet geladen worden. Misschien is deze verplaatst of verwijderd</string>
     <string name="unknown_name">Onbekende naam</string>
     <string name="unknown_error">Onbekende foutmelding</string>
-    <string name="insert_an_sd_card">Plaats een SD kaart om afbeeldingen toe te voegen</string>
-    <string name="please_grant_notally">Geef Notally rechten om notificaties te versturen. Dit wordt gebruikt om de voortgang weer te geven wanneer bijvoorbeeld verwijderen afbeeldingen of importeren van backups langer duurt</string>
 
     <plurals name="cant_add_images">
         <item quantity="one">Kan %d afbeelding niet toevoegen</item>
diff --git a/gradlew b/gradlew
old mode 100644
new mode 100755
