diff --git a/ui/build.gradle b/ui/build.gradle
index 5e21010..dd126d4 100644
--- a/ui/build.gradle
+++ b/ui/build.gradle
@@ -66,6 +66,17 @@ android {
         disable 'LongLogTag'
         warning 'MissingTranslation', 'ImpliedQuantity'
     }
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
