diff --git a/ui/build.gradle b/ui/build.gradle
index 6c58dec..05b01d7 100644
--- a/ui/build.gradle
+++ b/ui/build.gradle
@@ -53,6 +53,12 @@ android {
     lintOptions {
         disable('LongLogTag')
     }
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
