diff --git a/ui/build.gradle b/ui/build.gradle
index 2bbbe53..7dbb799 100644
--- a/ui/build.gradle
+++ b/ui/build.gradle
@@ -60,6 +60,12 @@ android {
         warning('MissingTranslation')
         warning('ImpliedQuantity')
     }
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
