diff --git a/OpenKeychain/build.gradle b/OpenKeychain/build.gradle
index 9a8b8ed55..a17d69700 100644
--- a/OpenKeychain/build.gradle
+++ b/OpenKeychain/build.gradle
@@ -263,5 +263,11 @@ android {
         exclude 'META-INF/NOTICE'
         exclude '.readme'
     }
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
