diff --git a/OpenKeychain/build.gradle b/OpenKeychain/build.gradle
index 8b4c4251d..91d4bfa15 100644
--- a/OpenKeychain/build.gradle
+++ b/OpenKeychain/build.gradle
@@ -290,6 +290,13 @@ android {
         exclude 'META-INF/NOTICE'
         exclude '.readme'
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 task jacocoTestReport(type:JacocoReport, dependsOn: "testFdroidDebugWithTestCoverageUnitTest") {
