diff --git a/OpenKeychain/build.gradle b/OpenKeychain/build.gradle
index 6465a9951..e9d561634 100644
--- a/OpenKeychain/build.gradle
+++ b/OpenKeychain/build.gradle
@@ -110,6 +110,13 @@ dependencies {
 
     // for debugging the db. don't enable by default, this will expose the database no the network!
     // debugImplementation 'com.amitshekhar.android:debug-db:1.0.3'
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 // Output of ./gradlew -q calculateChecksums
