diff --git a/app/build.gradle b/app/build.gradle
index 4ccd349..2beac2a 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -22,7 +22,7 @@ android {
 	splits {
 		abi {
 			enable true
-			universalApk true
+			universalApk false
 		}
 	}
 
@@ -50,4 +50,16 @@ android {
 			}
 		}
 	}
+
+	Calendar origin = Calendar.getInstance()
+	origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+	Date d1 = origin.getTime()
+	Date d2 = new Date()
+	long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+	def DateVersionCode = 2112312359 + minutes
+	android.applicationVariants.all { variant ->
+		variant.outputs.all { output ->
+			output.versionCodeOverride = DateVersionCode
+		}
+	}
 }
diff --git a/jni b/jni
index ca3182b..014a367 160000
--- a/jni
+++ b/jni
@@ -1 +1 @@
-Subproject commit ca3182bcde76289a713b03ce03de9f6d426f941c
+Subproject commit 014a36787a195b85dd5def12845b7e9a1f06a07d
