diff --git a/app/build.gradle b/app/build.gradle
index b6dc41d..af127fd 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -23,7 +23,7 @@ android {
 	splits {
 		abi {
 			enable true
-			universalApk true
+			universalApk false
 		}
 	}
 
@@ -48,4 +48,12 @@ android {
 			}
 		}
 	}
+
+	def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+	project.ext.versionCodes = ['armeabi-v7a': 1, 'arm64-v8a': 2, 'x86': 3, 'x86_64': 4]
+	android.applicationVariants.all { variant ->
+		variant.outputs.all { output ->
+			output.versionCodeOverride = project.ext.versionCodes.get(output.getFilter(com.android.build.OutputFile.ABI), 0) + DateVersionCode
+		}
+	}
 }
diff --git a/jni b/jni
--- a/jni
+++ b/jni
@@ -1 +1 @@
-Subproject commit 88714df8c7e0ae4b7d009fc7ce6e10885d31aaf6
+Subproject commit 88714df8c7e0ae4b7d009fc7ce6e10885d31aaf6-dirty
