diff --git a/app/build.gradle b/app/build.gradle
index fa7ea56..7f127f6 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -22,7 +22,7 @@ android {
 	splits {
 		abi {
 			enable true
-			universalApk true
+			universalApk false
 		}
 	}
 
@@ -50,4 +50,16 @@ android {
 			}
 		}
 	}
+
+	Calendar origin = Calendar.getInstance()
+	origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+	Date d1 = origin.getTime()
+	Date d2 = new Date()
+	long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+	def DateVersionCode = 2112312359 + minutes
+	android.applicationVariants.all { variant ->
+		variant.outputs.all { output ->
+			output.versionCodeOverride = DateVersionCode
+		}
+	}
 }
