diff --git a/app/build.gradle b/app/build.gradle
index b6dc41d..6ff76a2 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -23,7 +23,7 @@ android {
 	splits {
 		abi {
 			enable true
-			universalApk true
+			universalApk false
 		}
 	}
 
@@ -48,4 +48,11 @@ android {
 			}
 		}
 	}
+
+	def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+	android.applicationVariants.all { variant ->
+		variant.outputs.all { output ->
+			output.versionCodeOverride = DateVersionCode
+		}
+	}
 }
diff --git a/jni b/jni
--- a/jni
+++ b/jni
@@ -1 +1 @@
-Subproject commit 88714df8c7e0ae4b7d009fc7ce6e10885d31aaf6
+Subproject commit 88714df8c7e0ae4b7d009fc7ce6e10885d31aaf6-dirty
