diff --git a/app/build.gradle b/app/build.gradle
index 816d21c..6b381e6 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -23,7 +23,7 @@ android {
 	splits {
 		abi {
 			enable true
-			universalApk true
+			universalApk false
 		}
 	}
 
@@ -48,4 +48,11 @@ android {
 			}
 		}
 	}
+
+	def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+	android.applicationVariants.all { variant ->
+		variant.outputs.all { output ->
+			output.versionCodeOverride = DateVersionCode
+		}
+    	}
 }
diff --git a/build.gradle b/build.gradle
index d691e44..6a9e9d9 100644
--- a/build.gradle
+++ b/build.gradle
@@ -1,7 +1,7 @@
 buildscript {
 	repositories {
-		jcenter()
 		google()
+		jcenter()
 	}
 	dependencies {
 		classpath 'com.android.tools.build:gradle:3.5.0'
@@ -13,8 +13,8 @@ allprojects {
 		if (project.hasProperty('MAVEN_REPO')) {
 			maven { url MAVEN_REPO }
 		}
-		jcenter()
 		google()
+		jcenter()
 		maven { url 'http://maven.ghostscript.com/' }
 	}
 }
