diff --git a/app/build.gradle b/app/build.gradle
index 40d4524..154352f 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -17,13 +17,13 @@ android {
 		minSdkVersion 16
 		targetSdkVersion 29
 		versionName '1.16.1'
-		versionCode 60
+		versionCode Integer.parseInt(new Date().format("yyMMddHHmm"))
 	}
 
 	splits {
 		abi {
 			enable true
-			universalApk true
+			universalApk false
 		}
 	}
 
diff --git a/build.gradle b/build.gradle
index d691e44..6a9e9d9 100644
--- a/build.gradle
+++ b/build.gradle
@@ -1,7 +1,7 @@
 buildscript {
 	repositories {
-		jcenter()
 		google()
+		jcenter()
 	}
 	dependencies {
 		classpath 'com.android.tools.build:gradle:3.5.0'
@@ -13,8 +13,8 @@ allprojects {
 		if (project.hasProperty('MAVEN_REPO')) {
 			maven { url MAVEN_REPO }
 		}
-		jcenter()
 		google()
+		jcenter()
 		maven { url 'http://maven.ghostscript.com/' }
 	}
 }
