diff --git a/build.gradle.kts b/build.gradle.kts
index 08db2b48..e6bbac25 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -2,6 +2,9 @@ import com.android.build.gradle.internal.tasks.factory.dependsOn
 import java.text.SimpleDateFormat
 import java.util.Date
 import java.util.TimeZone
+import com.android.build.api.variant.FilterConfiguration.FilterType.*
+import java.time.LocalDateTime
+import java.time.Duration;
 
 buildscript {
     dependencies {
@@ -241,6 +244,18 @@ android {
         checkReleaseBuilds = false
         disable += listOf("MissingTranslation", "ExtraTranslation")
     }
+
+    val d1 = LocalDateTime.of(2021, 12, 31, 23, 59);
+    val d2 = LocalDateTime.now()
+    val minutes = Duration.between(d1, d2).getSeconds() / 60
+    val DateVersionCode = (2112312359 + minutes).toInt()
+    androidComponents {
+        onVariants { variant ->
+            variant.outputs.forEach { output ->
+                output.versionCode.set(DateVersionCode)
+            }
+        }
+    }
 }
 
 dependencies {
@@ -365,4 +380,4 @@ fun getBuildDate(): String {
     val RFC3339_LIKE = SimpleDateFormat("yyyy-MM-dd'T'HH:mm'Z'")
     RFC3339_LIKE.timeZone = TimeZone.getTimeZone("UTC")
     return RFC3339_LIKE.format(Date())
-}
\ No newline at end of file
+}
diff --git a/gradlew b/gradlew
old mode 100644
new mode 100755
