diff --git a/build.gradle b/build.gradle
index 0cb6160..d94dd0f 100644
--- a/build.gradle
+++ b/build.gradle
@@ -60,7 +60,6 @@ android {
       debuggable false
       proguardFiles getDefaultProguardFile('proguard-android-optimize.txt')
       resValue "string", "app_name", "@string/app_name_release"
-      signingConfig signingConfigs.release
     }
 
     debug {
@@ -70,7 +69,6 @@ android {
       applicationIdSuffix ".debug"
       resValue "string", "app_name", "@string/app_name_debug"
       resValue "bool", "debug_logs", "true"
-      signingConfig signingConfigs.debug
     }
   }
 
@@ -90,6 +88,18 @@ android {
     // Translation are already checked by 'syncTranslations'
     disable 'MissingTranslation'
   }
+
+  Calendar origin = Calendar.getInstance()
+  origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+  Date d1 = origin.getTime()
+  Date d2 = new Date()
+  long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+  def DateVersionCode = 2112312359 + minutes
+  android.applicationVariants.all { variant ->
+      variant.outputs.all { output ->
+          output.versionCodeOverride = DateVersionCode
+      }
+  }
 }
 
 tasks.register('buildKeyboardFont') {
