diff --git a/app/build.gradle.kts b/app/build.gradle.kts
index f31010df..6e79a08b 100644
--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -2,6 +2,9 @@
 
 import org.jetbrains.kotlin.gradle.dsl.JvmTarget
 
+import com.android.build.api.variant.FilterConfiguration.FilterType.*
+import java.time.LocalDateTime
+import java.time.Duration
 
 plugins {
     id("com.android.application")
@@ -39,6 +42,11 @@ android {
         vectorDrawables {
             useSupportLibrary = true
         }
+        ndk {
+            abiFilters.apply {
+                add("arm64-v8a")
+            }
+        }
     }
 
     lint {
@@ -78,10 +86,6 @@ android {
 
     buildTypes {
         release {
-            if (project.hasProperty("RELEASE_STORE_FILE")) {
-                signingConfig = signingConfigs.getByName("release")
-            }
-
             // Keep using until AGP 9.0 then research proguard rules to retain debug info for stack traces
             postprocessing {
                 isRemoveUnusedCode = true
@@ -105,6 +109,17 @@ android {
     }
     // Match build tools with CI image
     buildToolsVersion = "36.0.0"
+    val d1 = LocalDateTime.of(2021, 12, 31, 23, 59);
+    val d2 = LocalDateTime.now()
+    val minutes = Duration.between(d1, d2).getSeconds() / 60
+    val DateVersionCode = (2112312359 + minutes).toInt()
+    androidComponents {
+        onVariants { variant ->
+            variant.outputs.forEach { output ->
+                output.versionCode.set(DateVersionCode)
+            }
+        }
+    }
 }
 
 ksp {
