diff --git a/app/build.gradle.kts b/app/build.gradle.kts
index 9fc39a9bd..2d2327a79 100644
--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -2,6 +2,10 @@
  * Copyright © All Contributors. See LICENSE and AUTHORS in the root directory for details.
  */
 
+import com.android.build.api.variant.FilterConfiguration.FilterType.*
+import java.time.LocalDateTime
+import java.time.Duration;
+
 plugins {
     alias(libs.plugins.android.application)
     alias(libs.plugins.compose.compiler)
@@ -30,6 +34,12 @@ android {
         buildConfigField("boolean", "customCertsUI", "true")
 
         testInstrumentationRunner = "at.bitfire.davdroid.HiltTestRunner"
+
+	ndk {
+            abiFilters.apply {
+                add("arm64-v8a")
+            }
+        }
     }
 
     java {
@@ -82,8 +92,6 @@ android {
             proguardFiles(getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules-release.pro")
 
             isShrinkResources = true
-
-            signingConfig = signingConfigs.findByName("bitfire")
         }
     }
 
@@ -116,6 +124,18 @@ android {
             }
         }
     }
+
+    val d1 = LocalDateTime.of(2021, 12, 31, 23, 59);
+    val d2 = LocalDateTime.now()
+    val minutes = Duration.between(d1, d2).getSeconds() / 60
+    val DateVersionCode = (2112312359 + minutes).toInt()
+    androidComponents {
+        onVariants { variant ->
+            variant.outputs.forEach { output ->
+                output.versionCode.set(DateVersionCode)
+            }
+        }
+    }
 }
 
 ksp {
