diff --git a/app/build.gradle b/app/build.gradle
index 5f9391c5..375fd2a4 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -93,8 +93,6 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules-release.pro'
 
             shrinkResources true
-
-            signingConfig signingConfigs.bitfire
         }
     }
 
@@ -111,6 +109,18 @@ android {
     androidResources {
         generateLocaleConfig true
     }
+
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 ksp {
@@ -215,4 +225,4 @@ dependencies {
 
     testImplementation "com.squareup.okhttp3:mockwebserver:${versions.okhttp}"
     testImplementation 'junit:junit:4.13.2'
-}
\ No newline at end of file
+}
