diff --git a/app/build.gradle b/app/build.gradle
index 2b2845ee..11800659 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -82,6 +82,17 @@ android {
     }
     namespace 'org.xbmc.kore'
 
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 java {
