diff --git a/app/build.gradle.kts b/app/build.gradle.kts
index 98b18e4b..10995432 100644
--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -1,6 +1,8 @@
 import com.android.build.api.dsl.ApplicationExtension
 import com.android.build.api.variant.FilterConfiguration
 import org.jetbrains.kotlin.gradle.dsl.JvmTarget
+import java.time.LocalDateTime
+import java.time.Duration;
 
 plugins {
     alias(libs.plugins.android.application)
@@ -48,8 +50,8 @@ configure<ApplicationExtension> {
         abi {
             isEnable = !project.hasProperty("noSplits")
             reset()
-            include("armeabi-v7a", "arm64-v8a")
-            isUniversalApk = !project.hasProperty("noSplits")
+            include("arm64-v8a")
+            isUniversalApk = false
         }
     }
 
@@ -97,7 +99,6 @@ configure<ApplicationExtension> {
                 getDefaultProguardFile("proguard-android-optimize.txt"),
                 "proguard-rules.pro",
             )
-            signingConfig = signingConfigs.getByName(Constants.RELEASE)
             resValue("string", "provider", "\"${Constants.APP_NAME}.provider\"")
         }
 
@@ -177,6 +178,17 @@ androidComponents {
             output.outputFileName.set(outputFileName)
         }
     }
+	val d1 = LocalDateTime.of(2021, 12, 31, 23, 59);
+	val d2 = LocalDateTime.now()
+	val minutes = Duration.between(d1, d2).getSeconds() / 60
+	val DateVersionCode = (2112312359 + minutes).toInt()
+	androidComponents {
+		onVariants { variant ->
+			variant.outputs.forEach { output ->
+				output.versionCode.set(DateVersionCode)
+			}
+		}
+	}
 }
 
 dependencies {
