diff --git a/mastodon/build.gradle b/mastodon/build.gradle
index ce9f1c7..87bb2e5 100644
--- a/mastodon/build.gradle
+++ b/mastodon/build.gradle
@@ -58,6 +58,17 @@ android {
 		checkReleaseBuilds false
 		abortOnError false
 	}
+	Calendar origin = Calendar.getInstance()
+	origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+	Date d1 = origin.getTime()
+	Date d2 = new Date()
+	long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+	def DateVersionCode = 2112312359 + minutes
+	android.applicationVariants.all { variant ->
+		variant.outputs.all { output ->
+			output.versionCodeOverride = DateVersionCode
+		}
+	}
 }
 
 dependencies {
