diff --git a/app/build.gradle.kts b/app/build.gradle.kts
index f5809ba5f..85c26143e 100644
--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -1,4 +1,8 @@
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+import com.android.build.api.variant.FilterConfiguration.FilterType.*
+import java.time.LocalDateTime
+import java.time.Duration;
+
 
 plugins {
     id("com.android.application")
@@ -37,10 +41,6 @@ android {
         // Please disable ACRA or use your own instance in forked versions of the project
         buildConfigField("String", "ACRA_URI", "\"https://tachiyomi.kanade.eu/crash_report\"")
 
-        ndk {
-            abiFilters += SUPPORTED_ABIS
-        }
-
         testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
     }
 
@@ -49,7 +49,7 @@ android {
             isEnable = true
             reset()
             include(*SUPPORTED_ABIS.toTypedArray())
-            isUniversalApk = true
+            isUniversalApk = false
         }
     }
 
@@ -136,6 +136,23 @@ android {
     kotlinOptions {
         jvmTarget = JavaVersion.VERSION_1_8.toString()
     }
+
+    val d1 = LocalDateTime.of(2021, 12, 31, 23, 59);
+    val d2 = LocalDateTime.now()
+    val minutes = Duration.between(d1, d2).getSeconds() / 60
+    val DateVersionCode = (2112312359 + minutes).toInt()
+    val abiCodes = mapOf("armeabi-v7a" to 1, "arm64-v8a" to 2, "x86" to 3, "x86_64" to 4)
+    androidComponents {
+        onVariants { variant ->
+            variant.outputs.forEach { output ->
+                val name = output.filters.find { it.filterType == ABI }?.identifier
+                val baseAbiCode = abiCodes[name]
+                if (baseAbiCode != null) {
+                    output.versionCode.set(baseAbiCode + DateVersionCode)
+                }
+            }
+        }
+    }
 }
 
 dependencies {
diff --git a/app/src/main/java/eu/kanade/tachiyomi/ui/browse/migration/sources/MigrationSourcesController.kt b/app/src/main/java/eu/kanade/tachiyomi/ui/browse/migration/sources/MigrationSourcesController.kt
index 136e418ac..c400c0926 100644
--- a/app/src/main/java/eu/kanade/tachiyomi/ui/browse/migration/sources/MigrationSourcesController.kt
+++ b/app/src/main/java/eu/kanade/tachiyomi/ui/browse/migration/sources/MigrationSourcesController.kt
@@ -52,13 +52,13 @@ class MigrationSourcesController : ComposeController<MigrationSourcesPresenter>(
             }
             R.id.asc_alphabetical,
             R.id.desc_alphabetical,
--> {
+            -> {
                 presenter.setAlphabeticalSorting(itemId == R.id.asc_alphabetical)
                 true
             }
             R.id.asc_count,
             R.id.desc_count,
--> {
+            -> {
                 presenter.setTotalSorting(itemId == R.id.asc_count)
                 true
             }
diff --git a/app/src/main/java/eu/kanade/tachiyomi/widget/TachiyomiBottomNavigationView.kt b/app/src/main/java/eu/kanade/tachiyomi/widget/TachiyomiBottomNavigationView.kt
index e3209c384..6ed163a2c 100644
--- a/app/src/main/java/eu/kanade/tachiyomi/widget/TachiyomiBottomNavigationView.kt
+++ b/app/src/main/java/eu/kanade/tachiyomi/widget/TachiyomiBottomNavigationView.kt
@@ -91,7 +91,8 @@ class TachiyomiBottomNavigationView @JvmOverloads constructor(
                     currentAnimator = null
                     postInvalidate()
                 }
-            },)
+            },
+            )
     }
 
     internal class SavedState : AbsSavedState {
