diff --git a/app/build.gradle b/app/build.gradle
index fda95086..8fa0c150 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -106,6 +106,13 @@ android {
     kotlinOptions {
         jvmTarget = "1.8"
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
