diff --git a/app/build.gradle b/app/build.gradle
index 3a062b60..d38ae74f 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -119,6 +119,13 @@ android {
     kotlinOptions {
         jvmTarget = JavaVersion.VERSION_1_8.toString()
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 androidExtensions {
