diff --git a/app/build.gradle.kts b/app/build.gradle.kts
index 8ce1863e..f63f4006 100644
--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -1,4 +1,5 @@
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+import com.android.build.gradle.internal.api.ApkVariantOutputImpl
 import java.io.ByteArrayOutputStream
 import java.text.SimpleDateFormat
 import java.util.Date
@@ -111,6 +112,13 @@ android {
     kotlinOptions {
         jvmTarget = JavaVersion.VERSION_1_8.toString()
     }
+
+    val DateVersionCode = SimpleDateFormat("yyMMddHHmm").format(Date()).toInt()
+    android.applicationVariants.all {
+        outputs.forEach { output ->
+            (output as ApkVariantOutputImpl).versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
