diff --git a/app/build.gradle b/app/build.gradle
index 302c1a54..caa3cf6c 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -109,6 +109,13 @@ android {
     kotlinOptions {
         jvmTarget = "1.8"
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 androidExtensions {
