diff --git a/app/build.gradle b/app/build.gradle
index d4d21670..e39267a3 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -100,6 +100,13 @@ android {
         disable 'InvalidPackage'
         abortOnError false
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
