diff --git a/komelia-app/build.gradle.kts b/komelia-app/build.gradle.kts
index a82676a..1246178 100644
--- a/komelia-app/build.gradle.kts
+++ b/komelia-app/build.gradle.kts
@@ -2,6 +2,8 @@ import org.jetbrains.compose.desktop.application.dsl.TargetFormat
 import org.jetbrains.kotlin.gradle.ExperimentalWasmDsl
 import org.jetbrains.kotlin.gradle.dsl.JvmTarget
 import org.jetbrains.kotlin.gradle.targets.js.webpack.KotlinWebpackConfig
+import java.time.LocalDateTime
+import java.time.Duration;
 
 plugins {
     alias(libs.plugins.kotlinMultiplatform)
@@ -86,6 +88,11 @@ android {
         targetSdk = libs.versions.android.targetSdk.get().toInt()
         versionCode = 9
         versionName = libs.versions.app.version.get()
+        ndk {
+            abiFilters.apply {
+                add("arm64-v8a")
+            }
+        }
     }
     packaging {
         resources {
@@ -110,6 +117,17 @@ android {
         sourceCompatibility = JavaVersion.VERSION_17
         targetCompatibility = JavaVersion.VERSION_17
     }
+    val d1 = LocalDateTime.of(2021, 12, 31, 23, 59);
+    val d2 = LocalDateTime.now()
+    val minutes = Duration.between(d1, d2).getSeconds() / 60
+    val DateVersionCode = (2112312359 + minutes).toInt()
+    androidComponents {
+        onVariants { variant ->
+            variant.outputs.forEach { output ->
+                output.versionCode.set(DateVersionCode)
+            }
+        }
+    }
 }
 
 
