diff --git a/app/build.gradle b/app/build.gradle
index ec2383e52..f25d5baa9 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -102,6 +102,18 @@ android {
     lint {
         abortOnError false
     }
+
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+            variant.outputs.all { output ->
+                    output.versionCodeOverride = DateVersionCode
+            }
+    }
 }
 
 dependencies {
