diff --git a/app/build.gradle b/app/build.gradle
index ca2d905e..6e5f517e 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -23,7 +23,7 @@ android {
         applicationId "com.amaze.filemanager"
         minSdkVersion 14
         targetSdkVersion 28
-        versionCode 65
+        versionCode Integer.parseInt(new Date().format("yyMMddHHmm"))
         versionName "3.3.0RC1"
         multiDexEnabled true
 
