diff --git a/app/build.gradle b/app/build.gradle
index 5d3e51e2..ea11ded5 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -84,6 +84,13 @@ android {
             returnDefaultValues = true
         }
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 ext {
