diff --git a/app/build.gradle b/app/build.gradle
index 1877916d9..a0e1d122c 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -105,6 +105,13 @@ android {
             returnDefaultValues = true
         }
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
