diff --git a/app/build.gradle b/app/build.gradle
index 0a5aa3407..861170cc5 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -94,6 +94,13 @@ android {
             returnDefaultValues = true
         }
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 ext {
