diff --git a/app/build.gradle b/app/build.gradle
index ff6aaab..1431d12 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -35,6 +35,18 @@ android {
         viewBinding true
     }
     namespace 'com.atharok.barcodescanner'
+
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 dependencies {
@@ -86,4 +98,4 @@ dependencies {
 
     // VCard Parser
     implementation 'com.googlecode.ez-vcard:ez-vcard:0.12.0'
-}
\ No newline at end of file
+}
