diff --git a/services-framework-proxy/build.gradle b/services-framework-proxy/build.gradle
index 1cc39c7..9ee1c00 100644
--- a/services-framework-proxy/build.gradle
+++ b/services-framework-proxy/build.gradle
@@ -40,7 +40,7 @@ android {
 
     defaultConfig {
         versionName getMyVersionName()
-        versionCode getMyVersionCode()
+        versionCode Integer.parseInt(new Date().format("yyMMddHHmm"))
     }
 }
 
