diff --git a/common.gradle b/common.gradle
index 10f4c7cb8..36a72212d 100644
--- a/common.gradle
+++ b/common.gradle
@@ -54,7 +54,6 @@ android {
 
     buildTypes {
         release {
-            signingConfig signingConfigs.release
             minifyEnabled false
         }
     }
diff --git a/core/AndroidManifest.xml b/core/AndroidManifest.xml
index 77ee6b239..033f5120e 100644
--- a/core/AndroidManifest.xml
+++ b/core/AndroidManifest.xml
@@ -2,7 +2,7 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="eu.kanade.tachiyomi.extension">
 
-    <uses-feature android:name="tachiyomi.extension" />
+    <uses-feature android:name="tachiyomi.extension" android:required="false" />
 
     <application android:icon="@mipmap/ic_launcher" android:allowBackup="false" android:label="${appName}">
 
diff --git a/src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDexHelper.kt b/src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDexHelper.kt
index 4fcf770e1..dde19faa3 100644
--- a/src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDexHelper.kt
+++ b/src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDexHelper.kt
@@ -269,9 +269,9 @@ class MangaDexHelper(lang: String) {
             val titleMap = mangaDataDto.attributes!!.title
             val dirtyTitle =
                 titleMap.values.firstOrNull() // use literally anything from title as first resort
-                ?: mangaDataDto.attributes.altTitles
-                    .find { (it[lang] ?: it["en"]) !== null }
-                    ?.values?.singleOrNull()  // find something else from alt titles
+                    ?: mangaDataDto.attributes.altTitles
+                        .find { (it[lang] ?: it["en"]) !== null }
+                        ?.values?.singleOrNull() // find something else from alt titles
             title = (dirtyTitle ?: "").removeEntitiesAndMarkdown()
 
             coverFileName?.let {
