diff --git a/common.gradle b/common.gradle
index fa4222513..08c0f92ea 100644
--- a/common.gradle
+++ b/common.gradle
@@ -55,7 +55,6 @@ android {
 
     buildTypes {
         release {
-            signingConfig signingConfigs.release
             minifyEnabled false
         }
     }
diff --git a/core/AndroidManifest.xml b/core/AndroidManifest.xml
index 77ee6b239..033f5120e 100644
--- a/core/AndroidManifest.xml
+++ b/core/AndroidManifest.xml
@@ -2,7 +2,7 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="eu.kanade.tachiyomi.extension">
 
-    <uses-feature android:name="tachiyomi.extension" />
+    <uses-feature android:name="tachiyomi.extension" android:required="false" />
 
     <application android:icon="@mipmap/ic_launcher" android:allowBackup="false" android:label="${appName}">
 
diff --git a/src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDex.kt b/src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDex.kt
index ea65289a7..266ae29bd 100644
--- a/src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDex.kt
+++ b/src/all/mangadex/src/eu/kanade/tachiyomi/extension/all/mangadex/MangaDex.kt
@@ -291,7 +291,7 @@ abstract class MangaDex(override val lang: String, val dexLang: String) :
         val listDto = helper.json.decodeFromString<ListDto>(response.body!!.string())
         val listDtoFiltered = listDto.data.relationships.filter { it.type != "Manga" }
         val amount = listDtoFiltered.count()
-        if (amount < 1){
+        if (amount < 1) {
             throw Exception("No Manga in List")
         }
         val minIndex = (page - 1) * MDConstants.mangaLimit
