diff --git a/common.gradle b/common.gradle
index a287167..f85038e 100644
--- a/common.gradle
+++ b/common.gradle
@@ -44,7 +44,6 @@ android {
 
     buildTypes {
         release {
-            signingConfig signingConfigs.release
             minifyEnabled false
         }
     }
diff --git a/core/AndroidManifest.xml b/core/AndroidManifest.xml
index b56b3b9..ba4d436 100644
--- a/core/AndroidManifest.xml
+++ b/core/AndroidManifest.xml
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android">
 
-    <uses-feature android:name="tachiyomi.extension" />
+    <uses-feature android:name="tachiyomi.extension" android:required="false" />
 
     <application android:icon="@mipmap/ic_launcher" android:allowBackup="false" android:label="${appName}">
 
diff --git a/src/all/komga/src/eu/kanade/tachiyomi/extension/all/komga/Komga.kt b/src/all/komga/src/eu/kanade/tachiyomi/extension/all/komga/Komga.kt
index 4f0adb5..9ce793c 100644
--- a/src/all/komga/src/eu/kanade/tachiyomi/extension/all/komga/Komga.kt
+++ b/src/all/komga/src/eu/kanade/tachiyomi/extension/all/komga/Komga.kt
@@ -21,7 +21,6 @@ import eu.kanade.tachiyomi.extension.all.komga.dto.ReadListDto
 import eu.kanade.tachiyomi.extension.all.komga.dto.SeriesDto
 import eu.kanade.tachiyomi.network.GET
 import eu.kanade.tachiyomi.network.asObservable
-import eu.kanade.tachiyomi.network.await
 import eu.kanade.tachiyomi.source.ConfigurableSource
 import eu.kanade.tachiyomi.source.UnmeteredSource
 import eu.kanade.tachiyomi.source.model.Filter
