diff --git a/app/build.gradle b/app/build.gradle
index d82a28436..3a8e59552 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -97,6 +97,13 @@ android {
     kotlinOptions {
         jvmTarget = "1.8"
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 def room_version = "2.2.6"
