diff --git a/app/build.gradle b/app/build.gradle
index 952b7da32..d7c91ebe8 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -91,6 +91,18 @@ android {
     kotlinOptions {
         jvmTarget = "1.8"
     }
+
+    Calendar origin = Calendar.getInstance()
+    origin.set(2021, Calendar.DECEMBER, 31, 23, 59)
+    Date d1 = origin.getTime()
+    Date d2 = new Date()
+    long minutes = (d2.getTime() - d1.getTime()) / 1000 / 60
+    def DateVersionCode = 2112312359 + minutes
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 def room_version = "2.4.3"
