diff --git a/app/build.gradle b/app/build.gradle
index 07a7f3d9..4e9c1ec9 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -88,6 +88,13 @@ android {
         targetCompatibility JavaVersion.VERSION_1_8
         sourceCompatibility JavaVersion.VERSION_1_8
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 def room_version = "2.2.5"
