diff --git a/app/build.gradle b/app/build.gradle
index 83dd3e54..f7721a73 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -89,6 +89,13 @@ android {
     kotlinOptions {
         jvmTarget = "1.8"
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 def room_version = "2.2.5"
