diff --git a/app/build.gradle b/app/build.gradle
index bc641d45..4177af87 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -83,6 +83,13 @@ android {
         targetCompatibility JavaVersion.VERSION_1_8
         sourceCompatibility JavaVersion.VERSION_1_8
     }
+
+    def DateVersionCode = Integer.parseInt(new Date().format("yyMMddHHmm"))
+    android.applicationVariants.all { variant ->
+        variant.outputs.all { output ->
+            output.versionCodeOverride = DateVersionCode
+        }
+    }
 }
 
 def spongycastleVersion = "1.58.0.0"
